--- "D:\\Development\\WoW Python Tools\\Output\\11.0.2.55763\\EvokerPreservation.lua"	2024-07-21 09:33:02.089085700 -1000
+++ "D:\\Development\\WoW Python Tools\\Output\\11.0.5.56865\\EvokerPreservation.lua"	2024-10-21 13:54:37.174764500 -1000
@@ -1,54 +1,53 @@
 -- EvokerPreservation.lua
--- July 2024
+-- October 2024
 
 local addon, ns = ...
 local Hekili = _G[ addon ]
 local class, state = Hekili.Class, Hekili.State
 
 local spec = Hekili:NewSpecialization( 1468 )
 
 -- Resources
 spec:RegisterResource( Enum.PowerType.Mana )
 spec:RegisterResource( Enum.PowerType.Essence )
 
 spec:RegisterTalents( {
     -- Evoker Talents
     aerial_mastery        = { 93352, 365933, 1 }, -- Hover gains $s1 additional charge.
     ancient_flame         = { 93271, 369990, 1 }, -- Casting Emerald Blossom or Verdant Embrace reduces the cast time of your next Living Flame by $375583s1%.
     attuned_to_the_dream  = { 93292, 376930, 2 }, -- Your healing done and healing received are increased by $s1%.
     blast_furnace         = { 93309, 375510, 1 }, -- Fire Breath's damage over time lasts $s1 sec longer.
     bountiful_bloom       = { 93291, 370886, 1 }, -- Emerald Blossom heals $s1 additional allies.
     cauterizing_flame     = { 93294, 374251, 1 }, -- Cauterize an ally's wounds, removing all Bleed, Poison, Curse, and Disease effects. Heals for $s2 upon removing any effect.
-    clobbering_sweep      = { 93296, 375443, 1 }, -- Tail Swipe's cooldown is reduced by ${$s1/-1000} sec.
+    clobbering_sweep      = { 103844, 375443, 1 }, -- Tail Swipe's cooldown is reduced by ${$s1/-60000} min.
     draconic_legacy       = { 93300, 376166, 1 }, -- Your Stamina is increased by $s1%.
     enkindled             = { 93295, 375554, 2 }, -- Living Flame deals $s1% more damage and healing.
     expunge               = { 93306, 365585, 1 }, -- Expunge toxins affecting an ally, removing all Poison effects.
     extended_flight       = { 93349, 375517, 2 }, -- Hover lasts ${$s1/1000} sec longer.
     exuberance            = { 93299, 375542, 1 }, -- While above 75% health, your movement speed is increased by $s1%.
     fire_within           = { 93345, 375577, 1 }, -- Renewing Blaze's cooldown is reduced by ${$s1/-1000} sec.
     foci_of_life          = { 93345, 375574, 1 }, -- Renewing Blaze restores you more quickly, causing damage you take to be healed back over $<newDur> sec.
     forger_of_mountains   = { 93270, 375528, 1 }, -- Landslide's cooldown is reduced by ${$s1/-1000} sec, and it can withstand $s2% more damage before breaking.
-    heavy_wingbeats       = { 93296, 368838, 1 }, -- Wing Buffet's cooldown is reduced by ${$s1/-1000} sec.
+    heavy_wingbeats       = { 103843, 368838, 1 }, -- Wing Buffet's cooldown is reduced by ${$s1/-60000} min.
     inherent_resistance   = { 93355, 375544, 2 }, -- Magic damage taken reduced by $s1%.
     innate_magic          = { 93302, 375520, 2 }, -- Essence regenerates $s1% faster.
     instinctive_arcana    = { 93310, 376164, 2 }, -- Your Magic damage done is increased by $s1%.
     landslide             = { 93305, 358385, 1 }, -- Conjure a path of shifting stone towards the target location, rooting enemies for $355689d. Damage may cancel the effect.
     leaping_flames        = { 93343, 369939, 1 }, -- Fire Breath causes your next Living Flame to strike 1 additional target per empower level.
     lush_growth           = { 93347, 375561, 2 }, -- Green spells restore $s1% more health.
     natural_convergence   = { 93312, 369913, 1 }, -- Disintegrate channels $s1% faster$?c3[ and Eruption's cast time is reduced by $s3%][].
     obsidian_bulwark      = { 93289, 375406, 1 }, -- Obsidian Scales has an additional charge.
     obsidian_scales       = { 93304, 363916, 1 }, -- Reinforce your scales, reducing damage taken by $s1%. Lasts $d.
     oppressing_roar       = { 93298, 372048, 1 }, -- Let out a bone-shaking roar at enemies in a cone in front of you, increasing the duration of crowd controls that affect them by $s2% in the next $d.$?s374346[; Removes $s1 Enrage effect from each enemy.][]
     overawe               = { 93297, 374346, 1 }, -- Oppressing Roar removes $372048s1 Enrage effect from each enemy, and its cooldown is reduced by ${$s3/-1000} sec.
     panacea               = { 93348, 387761, 1 }, -- Emerald Blossom and Verdant Embrace instantly heal you for $387763s1 when cast.
-    permeating_chill      = { 93303, 370897, 1 }, -- Your damaging Blue spells reduce the target's movement speed by $s1% for $370898d.
     potent_mana           = { 93715, 418101, 1 }, -- Source of Magic increases the target's healing and damage done by $s1%.
     protracted_talons     = { 93307, 369909, 1 }, -- Azure Strike damages $s1 additional $Lenemy:enemies;.
     quell                 = { 93311, 351338, 1 }, -- Interrupt an enemy's spellcasting and prevent any spell from that school of magic from being cast for $d.
     recall                = { 93301, 371806, 1 }, -- You may reactivate $?s359816[Dream Flight and ][]$?s403631[Breath of Eons][Deep Breath] within $s1 sec after landing to travel back in time to your takeoff location.
     regenerative_magic    = { 93353, 387787, 1 }, -- Your Leech is increased by $s1%.
     renewing_blaze        = { 93354, 374348, 1 }, -- The flames of life surround you for $d. While this effect is active, $s1% of damage you take is healed back over $374349d.
     rescue                = { 93288, 370665, 1 }, -- Swoop to an ally and fly with them to the target location.
     scarlet_adaptation    = { 93340, 372469, 1 }, -- Store $s1% of your effective healing, up to $<cap>. Your next damaging Living Flame consumes all stored healing to increase its damage dealt.
     sleep_walk            = { 93293, 360806, 1 }, -- Disorient an enemy for $d, causing them to sleep walk towards you. Damage has a chance to awaken them.
     source_of_magic       = { 93344, 369459, 1 }, -- Redirect your excess magic to a friendly healer for $d. When you cast an empowered spell, you restore ${$372571s1/100}.2% of their maximum mana per empower level. Limit 1.
@@ -93,21 +92,21 @@
     flow_state            = { 93256, 385696, 2 }, -- Empower spells cause time to flow $s1% faster for you, increasing movement speed, cooldown recharge rate, and cast speed. Lasts $390148d.
     fluttering_seedlings  = { 93247, 359793, 2 }, -- Emerald Blossom sends out flying seedlings when it bursts, healing $s1 $Lally:allies; up to $s2 yds away for $361361s1.
     font_of_magic         = { 93252, 375783, 1 }, -- Your empower spells' maximum level is increased by 1.
     golden_hour           = { 93255, 378196, 1 }, -- Reversion instantly heals the target for $s1% of damage taken in the last $s2 sec.
     golden_opportunity    = { 94942, 432004, 1 }, -- $?c2[Casting Echo has a $s1% chance to cause your next Echo to be $s2% more effective.][Prescience has a $s1% chance to cause your next Prescience to last $s2% longer.]
     grace_period          = { 93265, 376239, 1 }, -- Your healing is increased by $s1% on targets with your Reversion.
     instability_matrix    = { 94930, 431484, 1 }, -- Each time you cast an empower spell, unstable time magic reduces its cooldown by up to $s1 sec.
     just_in_time          = { 93335, 376204, 1 }, -- Time Dilation's cooldown is reduced by $s1 sec each time you cast an Essence ability.
     lifebind              = { 93253, 373270, 1 }, -- Verdant Embrace temporarily bonds your life with an ally, causing your healing on either partner to heal the other for $s1% of the amount. Lasts $373267d.
     lifecinders           = { 94931, 444322, 1 }, -- Renewing Blaze also applies to your target or $s1 nearby injured $Lally:allies; at $s2% value.
-    lifeforce_mender      = { 93236, 376179, 2 }, -- Living Flame and Fire Breath deal additional damage and healing equal to $s1% of your maximum health.
+    lifeforce_mender      = { 93236, 376179, 2 }, -- Living Flame and Fire Breath deal additional damage and healing equal to ${$s1}.1% of your maximum health.
     lifegivers_flame      = { 93237, 371426, 1 }, -- Fire Breath heals $s3 nearby injured allies for $s1% of damage done to up to $s2 targets, split evenly among them.
     lifespark             = { 99804, 443177, 1 }, -- Reversion healing has a chance to cause your next Living Flame to cast instantly and deal $s2% increased healing or damage. Stacks up to $394552u charges.
     master_of_destiny     = { 94930, 431840, 1 }, -- Casting Essence spells extends all your active Threads of Fate by $s1 sec.
     motes_of_acceleration = { 94935, 432008, 1 }, -- Warp leaves a trail of Motes of Acceleration. Allies who come in contact with a mote gain 20% increased movement speed for 30 sec.
     nozdormus_teachings   = { 93258, 376237, 1 }, -- Temporal Anomaly reduces the cooldowns of your empower spells by $s1 sec.
     ouroboros             = { 93251, 381921, 1 }, -- Casting Echo grants one stack of Ouroboros, increasing the healing of your next Emerald Blossom by $387350s1%, stacking up to $387350u times.
     power_nexus           = { 93249, 369908, 1 }, -- Increases your maximum Essence to $<max>.
     primacy               = { 94951, 431657, 1 }, -- For each $?c2[healing over time effect from Spiritbloom][damage over time effect from Upheaval], gain $s1% haste, up to $s2%.
     punctuality           = { 93260, 371270, 1 }, -- Reversion has ${$s1+1} charges.
     red_hot               = { 94945, 444081, 1 }, -- Engulf gains $s2 additional charge and deals $s1% increased damage and healing.
@@ -126,21 +125,21 @@
     temporal_artificer    = { 93264, 381922, 1 }, -- Rewind's cooldown is reduced by ${$s1/-1000} sec.
     temporal_burst        = { 94955, 431695, 1 }, -- Tip the Scales overloads you with temporal energy, increasing your haste, movement speed, and cooldown recovery rate by ${$431698u*$431698s1}%, decreasing over $431698d.
     temporal_compression  = { 93241, 362874, 1 }, -- Each cast of a Bronze spell causes your next empower spell to reach maximum level in $s1% less time, stacking up to $362877u times.
     temporality           = { 94935, 431873, 1 }, -- Warp reduces damage taken by ${$s1/-1}%, starting high and reducing over $431872d.
     threads_of_fate       = { 94947, 431715, 1 }, -- Casting an empower spell during Temporal Burst causes a nearby ally to gain a Thread of Fate for $431716d, granting them a chance to echo their damage or healing spells, dealing $431716s1% of the amount again.
     time_convergence      = { 94932, 431984, 1 }, -- Non-defensive abilities with a $s1 second or longer cooldown grant $431991s1% Intellect for $431991d.; Essence spells extend the duration by $s2 sec.
     time_dilation         = { 93336, 357170, 1 }, -- Stretch time around an ally for the next $d, causing $s1% of damage they would take to instead be dealt over $d.$?s363899[; Healing a target under the effects of Time Dilation clears a portion of the delayed damage.][]
     time_lord             = { 93254, 372527, 2 }, -- Echo replicates $s1% more healing.
     time_of_need          = { 93259, 368412, 1 }, -- When you or an ally fall below $s1% health, a version of yourself enters your timeline and heals them for $361195s1. Your alternate self continues healing for $368415d before returning to their timeline.; May only occur once every $s2 sec.
     timeless_magic        = { 93263, 376240, 2 }, -- Reversion, Time Dilation, Echo, and Temporal Anomaly last $s1% longer and cost $s2% less mana.
-    titanic_precision     = { 94920, 445625, 1 }, -- Living Flame and Azure Strike have $s1 extra chance to trigger Essence Burst when they critically strike.
+    titanic_precision     = { 94920, 445625, 1 }, -- Living Flame and $?c1[Azure Strike][Reversion] have $s1 extra chance to trigger Essence Burst when they critically strike.
     titans_gift           = { 99803, 443264, 1 }, -- Essence Burst increases the effectiveness of your next Essence ability by $s1%.
     trailblazer           = { 94937, 444849, 1 }, -- $?c1[Hover and Deep Breath][Hover, Deep Breath, and Dream Flight] travel $s1% faster, and Hover travels $s1% further.
     traveling_flame       = { 99857, 444140, 1 }, -- Engulf increases the duration of $?c1[Fire Breath][Fire Breath or Dream Breath] by $s1 sec and causes it to spread to a target within $?c1[$s2][$s3] yds.
     warp                  = { 94948, 429483, 1 }, -- Hover now causes you to briefly warp out of existence and appear at your destination. Hover's cooldown is also reduced by ${$s1/-1000} sec.; Hover continues to allow Evoker spells to be cast while moving.
 } )
 
 -- PvP Talents
 spec:RegisterPvpTalents( {
     chrono_loop          = 5455, -- (383005) Trap the enemy in a time loop for $d. Afterwards, they are returned to their previous location and health. Cannot reduce an enemy's health below $s1%.
     divide_and_conquer   = 5595, -- (384689) $?s403631[Breath of Eons][Deep Breath] forms curtains of fire, preventing line of sight to enemies outside its walls and burning enemies who walk through them for $403516o1 Fire damage. Lasts $403727d.
@@ -231,21 +230,20 @@
     divide_and_conquer = {
         id = 403516,
         duration = 6.0,
         tick_time = 3.0,
         max_stack = 1,
     },
     -- Healing $w1 every $t1 sec.
     dream_breath = {
         id = 355941,
         duration = 2.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- lush_growth[375561] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- lush_growth[375561] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- spatial_paradox[406732] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
@@ -370,20 +368,21 @@
 
         -- Affected by:
         -- burning_adrenaline[444019] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'attributes': ['Suppress Points Stacking'], 'points': -30.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- flow_state[390148] #8: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- flow_state[390148] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
     },
     -- Sharing healing with an ally.
     lifebind = {
         id = 373267,
         duration = 5.0,
+        tick_time = 0.2,
         max_stack = 1,
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
     -- Next Living Flame's cast time reduced by $s1% and deals $s2% increased damage or healing.
     lifespark = {
@@ -443,23 +442,20 @@
     ouroboros = {
         id = 387350,
         duration = 3600,
         max_stack = 1,
     },
     -- Movement speed reduced by $w1%.
     permeating_chill = {
         id = 370898,
         duration = 3.0,
         max_stack = 1,
-
-        -- Affected by:
-        -- permeating_chill[370897] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
     },
     -- Restoring $w1 health every $t1 sec.
     renewing_blaze = {
         id = 374349,
         duration = 8.0,
         max_stack = 1,
 
         -- Affected by:
         -- foci_of_life[375574] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -4000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
@@ -469,21 +465,20 @@
         duration = 1.0,
         max_stack = 1,
 
         -- Affected by:
         -- spatial_paradox[406732] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
     -- Healing for $w1 every $t1 sec.
     reversion = {
         id = 366155,
         duration = 12.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- grace_period[376239] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- punctuality[371270] #0: { 'type': APPLY_AURA, 'subtype': MOD_MAX_CHARGES, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
         -- timeless_magic[376240] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
@@ -553,20 +548,30 @@
     },
     -- About to be grabbed!
     swoop_up = {
         id = 370388,
         duration = 1.0,
         max_stack = 1,
 
         -- Affected by:
         -- spatial_paradox[406732] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
+    -- Movement speed reduced by $s2%.
+    tail_swipe = {
+        id = 368970,
+        duration = 4.0,
+        max_stack = 1,
+
+        -- Affected by:
+        -- clobbering_sweep[375443] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -120000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
+        -- walloping_blow[387341] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
+    },
     -- Haste, movement speed, and cooldown recovery rate increased by $s1%.
     temporal_burst = {
         id = 431698,
         duration = 30.0,
         tick_time = 1.0,
         max_stack = 1,
     },
     -- Empower spells reach maximum level in $s1% less time.$?e1[ Healing of next empower spell increased by $s2%.][]
     temporal_compression = {
         id = 362877,
@@ -675,38 +680,38 @@
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.009,
         spendType = 'mana',
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 1.0879, 'pvp_multiplier': 1.68, 'variance': 0.05, 'radius': 6.0, 'target': TARGET_DEST_TARGET_ENEMY, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
+        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 1.08790004253, 'pvp_multiplier': 1.67999994755, 'variance': 0.05000000075, 'radius': 6.0, 'target': TARGET_DEST_TARGET_ENEMY, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
         -- #2: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 355627, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- preservation_evoker[356810] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- protracted_talons[369909] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
     },
 
     -- Weave the threads of time, reducing the cooldown of a major movement ability for all party and raid members by $381732s1% for $381732d.
     blessing_of_the_bronze = {
         id = 364342,
         color = 'bronze',
         cast = 0.0,
         cooldown = 15.0,
         gcd = "global",
 
-        spend = 0.040,
+        spend = 0.010,
         spendType = 'mana',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 381732, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 381741, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 381746, 'target': TARGET_UNIT_CASTER, }
         -- #3: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 381748, 'target': TARGET_UNIT_CASTER, }
         -- #4: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 381749, 'target': TARGET_UNIT_CASTER, }
@@ -786,32 +791,31 @@
         -- #5: { 'type': APPLY_AURA, 'subtype': MOD_MOVEMENT_FORCE_MAGNITUDE, 'points': -100.0, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- spatial_paradox[406732] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
 
     -- Tear into an enemy with a blast of blue magic, inflicting $o1 Spellfrost damage over $d, and slowing their movement speed by $370898s1% for $370898d.
     disintegrate = {
         id = 356995,
         color = 'blue',
-        cast = 3.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 3,
         spendType = 'essence',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DAMAGE, 'tick_time': 1.0, 'sp_bonus': 1.441, 'chain_targets': 1, 'pvp_multiplier': 1.2, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DAMAGE, 'tick_time': 1.0, 'sp_bonus': 1.58510005474, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- preservation_evoker[356810] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- preservation_evoker[356810] #16: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- hover[358267] #4: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- natural_convergence[369913] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- natural_convergence[369913] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- spatial_paradox[406732] #2: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- energy_loop[372233] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 35.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
@@ -859,23 +863,23 @@
     -- [355936] Inhale, gathering the power of the Dream. Release to exhale, healing $s1 injured allies in a 30 yd cone in front of you for ${$355941s3+$355941o1*8*$<ysera>}.; I:   Heals $355941s3 instantly and ${$355941o1*8*$<ysera>} over $<sOneDur> sec.; II:  Heals ${$355941s3+$355941o1*2*$<ysera>} instantly and ${$355941o1*6*$<ysera>} over $<sTwoDur> sec.; III: Heals ${$355941s3+$355941o1*4*$<ysera>} instantly and ${$355941o1*4*$<ysera>} over $<sThreeDur> sec.
     dream_breath_355941 = {
         id = 355941,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 2.0, 'sp_bonus': 0.384, 'radius': 30.0, 'target': TARGET_UNIT_CONE_ALLY, 'target2': TARGET_UNIT_DEST_AREA_ALLY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 2.0, 'sp_bonus': 0.38400000334, 'radius': 30.0, 'target': TARGET_UNIT_CONE_ALLY, 'target2': TARGET_UNIT_DEST_AREA_ALLY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, }
-        -- #2: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 0.768, 'radius': 30.0, 'target': TARGET_UNIT_CONE_ALLY, 'target2': TARGET_UNIT_DEST_AREA_ALLY, }
+        -- #2: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 0.76800000668, 'radius': 30.0, 'target': TARGET_UNIT_CONE_ALLY, 'target2': TARGET_UNIT_DEST_AREA_ALLY, }
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- lush_growth[375561] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- lush_growth[375561] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- spatial_paradox[406732] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
         -- expanded_lungs[444845] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
@@ -951,21 +955,21 @@
         spend = 2,
         spendType = 'essence',
 
         spend = 0.019,
         spendType = 'mana',
 
         talent = "echo",
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 1.2, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 1.20000004768, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 70.0, 'target': TARGET_UNIT_TARGET_ALLY, }
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- hover[358267] #4: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- spatial_paradox[406732] #2: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- time_lord[372527] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 50.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
@@ -1123,21 +1127,22 @@
         -- burning_adrenaline[444019] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'points': -30.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- flow_state[390148] #8: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- flow_state[390148] #9: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- flow_state[390148] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- temporal_compression[362877] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
 
     -- Increases haste by $s1% for all party and raid members for $d.; Allies receiving this effect will become Exhausted and unable to benefit from Fury of the Aspects or similar effects again for $57723d.
     fury_of_the_aspects = {
         id = 390386,
-        cast = 0.0,
+        cast = 40.0,
+        channeled = true,
         cooldown = 300.0,
         gcd = "none",
 
         spend = 0.040,
         spendType = 'mana',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': MELEE_SLOW, 'points': 30.0, 'radius': 50000.0, 'target': TARGET_UNIT_CASTER_AREA_RAID, }
@@ -1333,54 +1338,53 @@
         color = 'black',
         cast = 0.0,
         cooldown = 120.0,
         gcd = "global",
 
         talent = "oppressing_roar",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #1: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 1, 'schools': ['physical'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #2: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 27, 'schools': ['physical', 'holy', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #3: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 2, 'schools': ['holy'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #4: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 5, 'schools': ['physical', 'fire'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #5: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 13, 'schools': ['physical', 'fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #6: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 24, 'schools': ['nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #7: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 17, 'schools': ['physical', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #8: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 7, 'schools': ['physical', 'holy', 'fire'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #9: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 30, 'schools': ['holy', 'fire', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #10: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #11: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 12, 'schools': ['fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #12: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 23, 'schools': ['physical', 'holy', 'fire', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #13: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 18, 'schools': ['holy', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #14: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 10, 'schools': ['holy', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #15: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 14, 'schools': ['holy', 'fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #16: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 9, 'schools': ['physical', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #17: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 3, 'schools': ['physical', 'holy'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
-        -- #18: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.6, 'points': 50.0, 'value': 26, 'schools': ['holy', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #1: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 1, 'schools': ['physical'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #2: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 27, 'schools': ['physical', 'holy', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #3: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 2, 'schools': ['holy'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #4: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 5, 'schools': ['physical', 'fire'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #5: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 13, 'schools': ['physical', 'fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #6: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 24, 'schools': ['nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #7: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 17, 'schools': ['physical', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #8: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 7, 'schools': ['physical', 'holy', 'fire'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #9: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 30, 'schools': ['holy', 'fire', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #10: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #11: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 12, 'schools': ['fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #12: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 23, 'schools': ['physical', 'holy', 'fire', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #13: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 18, 'schools': ['holy', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #14: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 10, 'schools': ['holy', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #15: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 14, 'schools': ['holy', 'fire', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #16: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 9, 'schools': ['physical', 'nature'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #17: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 3, 'schools': ['physical', 'holy'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #18: { 'type': APPLY_AURA, 'subtype': MECHANIC_DURATION_MOD, 'pvp_multiplier': 0.60000002384, 'points': 50.0, 'value': 26, 'schools': ['holy', 'nature', 'frost'], 'radius': 30.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
 
         -- Affected by:
         -- overawe[374346] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -30000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
         -- spatial_paradox[406732] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
     },
 
     -- Interrupt an enemy's spellcasting and prevent any spell from that school of magic from being cast for $d.
     quell = {
         id = 351338,
         cast = 0.0,
         cooldown = 40.0,
         gcd = "none",
 
         talent = "quell",
         startsCombat = true,
-        interrupt = true,
 
         -- Effects:
         -- #0: { 'type': INTERRUPT_CAST, 'subtype': NONE, 'mechanic': interrupted, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- burning_adrenaline[444019] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'attributes': ['Suppress Points Stacking'], 'points': -30.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- flow_state[390148] #8: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- flow_state[390148] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
     },
 
@@ -1429,52 +1433,53 @@
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.028,
         spendType = 'mana',
 
         talent = "reversion",
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'attributes': ['Suppress Points Stacking'], 'tick_time': 2.0, 'sp_bonus': 0.342, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'attributes': ['Suppress Points Stacking'], 'tick_time': 2.0, 'sp_bonus': 0.34200000763, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': UNKNOWN, 'target': TARGET_UNIT_TARGET_ALLY, }
 
         -- Affected by:
         -- preservation_evoker[356810] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- attuned_to_the_dream[376930] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- grace_period[376239] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- punctuality[371270] #0: { 'type': APPLY_AURA, 'subtype': MOD_MAX_CHARGES, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
         -- timeless_magic[376240] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- timeless_magic[376240] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
     -- Rewind $?s376210[${$s3/2}%][$s3%] of damage taken in the last $s2 seconds by all allies within $a1 yds. Always heals for at least $<min>.; Healing increased by $s4% when not in a raid.
     rewind = {
         id = 363534,
         color = 'bronze',
-        cast = 0.0,
+        cast = 4.0,
+        channeled = true,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.055,
         spendType = 'mana',
 
         talent = "rewind",
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 1.0, 'sp_bonus': 0.4, 'radius': 40.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ALLY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 1.0, 'sp_bonus': 0.40000000596, 'radius': 40.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ALLY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, }
-        -- #2: { 'type': DUMMY, 'subtype': NONE, 'pvp_multiplier': 1.52, }
+        -- #2: { 'type': DUMMY, 'subtype': NONE, 'pvp_multiplier': 1.51999998093, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, }
 
         -- Affected by:
         -- spatial_paradox[406732] #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
         -- erasure[376210] #0: { 'type': APPLY_AURA, 'subtype': MOD_MAX_CHARGES, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Disorient an enemy for $d, causing them to sleep walk towards you. Damage has a chance to awaken them.
     sleep_walk = {
         id = 360806,
@@ -1548,22 +1553,21 @@
         -- #10: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
 
         -- Affected by:
         -- spatial_paradox[406732] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
 
     -- Divert spiritual energy, healing an ally for $367230s1. Splits to injured allies within $<range> yds when empowered.; I:   Heals one ally.; II:  Heals a second ally.; III: Heals a third ally.
     spiritbloom = {
         id = 367226,
         color = 'green',
-        cast = 2.5,
-        channeled = true,
+        cast = empowered_cast_time,
         empowered = true,
         cooldown = 30.0,
         gcd = "global",
 
         spend = 0.042,
         spendType = 'mana',
 
         talent = "spiritbloom",
         startsCombat = false,
 
@@ -1611,37 +1615,57 @@
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_DEST_DEST, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_ROOT, 'points': 5.0, 'target': TARGET_UNIT_TARGET_ANY, }
 
         -- Affected by:
         -- spatial_paradox[406732] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
 
+    -- Lash out with your tail, knocking enemies within $A1 yds into the air.
+    tail_swipe = {
+        id = 368970,
+        color = 'racial',
+        cast = 0.0,
+        cooldown = 180.0,
+        gcd = "global",
+
+        startsCombat = true,
+
+        -- Effects:
+        -- #0: { 'type': KNOCK_BACK_DEST, 'subtype': NONE, 'points': 110.0, 'value': 20, 'schools': ['fire', 'frost'], 'radius': 8.0, 'target': TARGET_DEST_CASTER, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
+        -- #1: { 'type': APPLY_AURA, 'subtype': MOD_DECREASE_SPEED, 'mechanic': snared, 'points': -70.0, 'radius': 8.0, 'target': TARGET_DEST_CASTER, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
+
+        -- Affected by:
+        -- clobbering_sweep[375443] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -120000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
+        -- walloping_blow[387341] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
+    },
+
     -- Send forward a vortex of temporal energy, absorbing $<shield> damage on you and any allies in its path. Absorption is reduced beyond $s2 targets.
     temporal_anomaly = {
         id = 373861,
         color = 'bronze',
         cast = 1.5,
         cooldown = 15.0,
         gcd = "global",
 
         spend = 0.075,
         spendType = 'mana',
 
         talent = "temporal_anomaly",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': CREATE_AREATRIGGER, 'subtype': NONE, 'points': 1.0, 'value': 25294, 'schools': ['holy', 'fire', 'nature', 'arcane'], 'radius': 0.0, 'target': TARGET_DEST_CASTER_FRONT, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
+        -- #2: { 'type': CREATE_AREATRIGGER, 'subtype': NONE, 'points': 1.0, 'value': 34997, 'schools': ['physical', 'fire', 'frost', 'shadow'], 'radius': 0.0, 'target': TARGET_DEST_CASTER_FRONT, }
 
         -- Affected by:
         -- preservation_evoker[356810] #8: { 'type': APPLY_AURA, 'subtype': MOD_COOLDOWN_BY_HASTE_REGEN, 'target': TARGET_UNIT_CASTER, }
         -- hover[358267] #4: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- spatial_paradox[406732] #2: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
         -- timeless_magic[376240] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- timeless_magic[376240] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
     -- Stretch time around an ally for the next $d, causing $s1% of damage they would take to instead be dealt over $d.$?s363899[; Healing a target under the effects of Time Dilation clears a portion of the delayed damage.][]
@@ -1734,21 +1758,21 @@
         cooldown = 9.0,
         gcd = "global",
 
         spend = 0.010,
         spendType = 'mana',
 
         talent = "unravel",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 7.205, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 9.0, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- preservation_evoker[356810] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- preservation_evoker[356810] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
 
     -- Fly to an ally and heal them for $361195s1, or heal yourself for the same amount.
     verdant_embrace = {
         id = 360995,
         color = 'green',
