--- "D:\\Development\\WoW Python Tools\\Output\\11.0.2.55763\\MonkWindwalker.lua"	2024-07-21 09:39:20.809734800 -1000
+++ "D:\\Development\\WoW Python Tools\\Output\\11.0.5.56865\\MonkWindwalker.lua"	2024-10-21 13:57:36.523047300 -1000
@@ -1,12 +1,12 @@
 -- MonkWindwalker.lua
--- July 2024
+-- October 2024
 
 local addon, ns = ...
 local Hekili = _G[ addon ]
 local class, state = Hekili.Class, Hekili.State
 
 local spec = Hekili:NewSpecialization( 269 )
 
 -- Resources
 spec:RegisterResource( Enum.PowerType.Energy )
 spec:RegisterResource( Enum.PowerType.Mana )
@@ -67,21 +67,21 @@
     winds_reach                    = { 101148, 450514, 1 }, -- The range of Disable is increased by $s1 yds.;  ; The duration of Crashing Momentum is increased by ${$s3/1000} sec and its snare now reduces movement speed by an additional $s2%.
     windwalking                    = { 101175, 157411, 1 }, -- You and your allies within $m2 yards have $s1% increased movement speed. Stacks with other similar effects.
     yulons_grace                   = { 101165, 414131, 1 }, -- Find resilience in the flow of chi in battle, gaining a magic absorb shield for ${$s1/10}.1% of your max health every $t sec in combat, stacking up to $s2%.
 
     -- Windwalker Talents
     acclamation                    = { 101036, 451432, 1 }, -- Rising Sun Kick increases the damage your target receives from you by $451433s1% for $451433d. Multiple instances may overlap.
     against_all_odds               = { 101253, 450986, 1 }, -- Flurry Strikes increase your Agility by $451061s1% for $451061d, stacking up to $451061u times.
     ascension                      = { 101037, 115396, 1 }, -- Increases your maximum Chi by $s1, maximum Energy by $s3, and your Energy regeneration by $s2%.
     august_dynasty                 = { 101235, 442818, 1 }, -- Casting Jadefire Stomp increases the $?c2[damage or healing of your next Rising Sun Kick by $442850s1% or Vivify by $442850s2%]?c3[damage of your next Rising Sun Kick by $442850s1%][].; This effect can only activate once every $proccooldown sec.
     brawlers_intensity             = { 101038, 451485, 1 }, -- The cooldown of Rising Sun Kick is reduced by ${$s1/-1000}.1 sec and the damage of Blackout Kick is increased by $s2%.
-    celestial_conduit              = { 101243, 443028, 1 }, -- $?c2[The August Celestials empower you, causing you to radiate ${$443039s1*$s7} healing onto up to $s3 injured allies and ${$443038s1*$s7} Nature damage onto enemies within $s6 yds over $d, split evenly among them. Healing and damage increased by $s1% per target, up to ${$s1*$s3}%.]?c3[The August Celestials empower you, causing you to radiate ${$443038s1*$s7} Nature damage onto enemies and ${$443039s1*$s7} healing onto up to $s3 injured allies within $443038A2 yds over $d, split evenly among them. Healing and damage increased by $s1% per enemy struck, up to ${$s1*$s3}%.][]; You may move while channeling, but casting other healing or damaging spells cancels this effect.; 
+    celestial_conduit              = { 101243, 443028, 1 }, -- $?c2[The August Celestials empower you, causing you to radiate ${$443039s1*$s7} healing onto up to $s3 injured allies and ${$443038s1*$s7} Nature damage onto enemies within $s6 yds over $d, split evenly among them. Healing and damage increased by $s1% per target, up to ${$s1*$s3}%.][The August Celestials empower you, causing you to radiate ${$443038s1*$s7} Nature damage onto enemies and ${$443039s1*$s7} healing onto up to $s3 injured allies within $443038A2 yds over $d, split evenly among them. Healing and damage increased by $s1% per enemy struck, up to ${$s1*$s3}%.]; You may move while channeling, but casting other healing or damaging spells cancels this effect.; 
     chi_burst                      = { 101159, 460485, 1 }, -- Your damaging spells and abilities have a chance to activate Chi Burst, allowing you to hurl a torrent of Chi energy up to $s1 yds forward, dealing $148135s1 Nature damage to all enemies, and $130654s1 healing to the Monk and all allies in its path. Healing and damage reduced beyond $123986s1 targets.; $?c1[; Casting Chi Burst does not prevent avoiding attacks.][]
     chi_wave                       = { 101159, 450391, 1 }, -- Every $t1 sec, your next Rising Sun Kick or Vivify releases a wave of Chi energy that flows through friends and foes, dealing $132467s1 Nature damage or $132463s1 healing. Bounces up to $115098s1 times to targets within $132466a2 yards.
     chijis_swiftness               = { 101240, 443566, 1 }, -- Your movement speed is increased by $s1% during Celestial Conduit and by $443569s1% for $443569d after being assisted by any Celestial.; 
     combat_wisdom                  = { 101217, 121817, 1 }, -- [451968] Expel negative chi from your body, healing for $s1 and dealing $s2% of the amount healed as Nature damage to an enemy within $115129A1 yards.$?s322102[; Draws in the positive chi of all your Healing Spheres to increase the healing of Expel Harm.][]$?s342928[; Generates ${$s3+$342928s2} Chi.][]
     communion_with_wind            = { 101041, 451576, 1 }, -- Strike of the Windlord's cooldown is reduced by ${$s1/-1000} sec and its damage is increased by $s2%.
     courage_of_the_white_tiger     = { 101242, 443087, 1 }, -- $?c2[Tiger Palm and Vivify have a chance to cause Xuen to claw a nearby enemy for $457917s1 Physical damage, healing a nearby ally for $s2% of the damage done.]?c3[Tiger Palm has a chance to cause Xuen to claw your target for $457917s1 Physical damage, healing a nearby ally for $s2% of the damage done.][Xuen claws your target for $457917s1 Physical damage, healing a nearby ally for $s2% of the damage done.]; $?c2[Invoke Yu'lon, the Jade Serpent or Invoke Chi-Ji, the Red Crane]?c3[Invoke Xuen, the White Tiger][Invoking a celestial] guarantees your next cast activates this effect.
     courageous_impulse             = { 101061, 451495, 1 }, -- The Blackout Kick! effect also increases the damage of your next Blackout Kick by $s1%.
     crane_vortex                   = { 101055, 388848, 1 }, -- Spinning Crane Kick damage increased by $s1% and its radius is increased by $s2%.
     dance_of_chiji                 = { 101060, 325201, 1 }, -- Spending Chi has a chance to make your next Spinning Crane Kick free and deal an additional $s1% damage.
     dance_of_the_wind              = { 101137, 432181, 1 }, -- Your dodge chance is increased by $s2% and an additional $432180s1% every $t1 sec until you dodge an attack, stacking up to $432180u times. 
@@ -135,21 +135,21 @@
     sequenced_strikes              = { 101059, 451515, 1 }, -- You have a $s1% chance to gain Blackout Kick! after consuming Dance of Chi-Ji.
     shadowboxing_treads            = { 101062, 392982, 1 }, -- Blackout Kick damage increased by $s2% and strikes an additional $s1 $ltarget;targets at $s3% effectiveness.
     singularly_focused_jade        = { 101043, 451573, 1 }, -- Jadefire Stomp's initial hit now strikes $s4 target, but deals $s2% increased damage and healing.
     spiritual_focus                = { 101052, 280197, 1 }, -- Every $s1 Chi you spend reduces the cooldown of $?s152173[Serenity][Storm, Earth, and Fire] by $?s152173[${$m3/1000}.1][${$m2/1000}.1] sec.
     storm_earth_and_fire           = { 101053, 137639, 1 }, -- Split into 3 elemental spirits for $d, each spirit dealing ${100+$m1}% of normal damage and healing.; You directly control the Storm spirit, while Earth and Fire spirits mimic your attacks on nearby enemies.; While active, casting Storm, Earth, and Fire again will cause the spirits to fixate on your target.
     strength_of_the_black_ox       = { 101241, 443110, 1 }, -- $?c2[After Xuen assists you, your next Enveloping Mist's cast time is reduced by $443112s1% and causes Niuzao to grant an absorb shield to $443113s3 nearby allies for $443113s2% of your maximum health.]?c3[After Xuen assists you, your next Blackout Kick refunds $s3 stacks of Teachings of the Monastery and causes Niuzao to stomp at your target's location, dealing $443127s1 damage to nearby enemies, reduced beyond $s2 targets.][]
     strike_of_the_windlord         = { 101215, 392983, 1 }, -- Strike with both fists at all enemies in front of you, dealing ${$395519s1+$395521s1} Physical damage and reducing movement speed by $s2% for $d.
     summon_white_tiger_statue      = { 101162, 450639, 1 }, -- Invoking Xuen, the White Tiger also spawns a White Tiger Statue at your location that pulses $389541s1 damage to all enemies every $s1 sec for $388686d.
     teachings_of_the_monastery     = { 101054, 116645, 1 }, -- Tiger Palm causes your next Blackout Kick to strike an additional time, stacking up to $202090u.; Blackout Kick has a $s1% chance to reset the remaining cooldown on Rising Sun Kick.; $?a210802[Each additional Blackout Kick restores ${$210803m1/100}.2% mana.][]
     temple_training                = { 101236, 442743, 1 }, -- $?c2[The healing of Enveloping Mist and Vivify is increased by $s1%.]?c3[Fists of Fury and Spinning Crane Kick deal $s2% more damage.][]
-    thunderfist                    = { 101040, 392985, 1 }, -- Strike of the Windlord grants you $s1 stacks of Thunderfist and an additional stack for each additional enemy struck. ; Thunderfist discharges upon melee strikes, dealing $242390s1 Nature damage.
+    thunderfist                    = { 101040, 392985, 1 }, -- Strike of the Windlord grants you $s1 stacks of Thunderfist and an additional stack for each additional enemy struck. ; Thunderfist discharges upon melee strikes, dealing $393566s1 Nature damage.
     touch_of_the_tiger             = { 101049, 388856, 1 }, -- Tiger Palm damage increased by $s1%.
     transfer_the_power             = { 101212, 195300, 1 }, -- Blackout Kick, Rising Sun Kick, and Spinning Crane Kick increase damage dealt by your next Fists of Fury by $s1%, stacking up to $195321u times.
     unity_within                   = { 101239, 443589, 1 }, -- Celestial Conduit can be recast once during its duration to call upon all of the August Celestials to assist you at $s1% effectiveness.; Unity Within is automatically cast when Celestial Conduit ends if not used before expiration.
     veterans_eye                   = { 101249, 450987, 1 }, -- Striking the same target $451071u times within $451071d grants $451085s1% Haste.; Multiple instances of this effect may overlap, stacking up to $451085u times.
     vigilant_watch                 = { 101244, 450993, 1 }, -- Blackout Kick deals an additional $s1% critical damage and increases the damage of your next set of Flurry Strikes by $451233s1%.
     whirling_dragon_punch          = { 101204, 152175, 1 }, -- Performs a devastating whirling upward strike, dealing ${3*$158221s1} damage to all nearby enemies and an additional $451767s1 damage to the first target struck. Damage reduced beyond $s1 targets.; Only usable while both Fists of Fury and Rising Sun Kick are on cooldown.
     whirling_steel                 = { 101245, 450991, 1 }, -- When your health drops below $s1%, summon Whirling Steel, increasing your parry chance and avoidance by $451214s1% for $451214d.; This effect can not occur more than once every $proccooldown sec.
     wisdom_of_the_wall             = { 101252, 450994, 1 }, -- [452684] Critical strike damage increased by $s1%.
     xuens_battlegear               = { 101210, 392993, 1 }, -- Rising Sun Kick critical strikes reduce the cooldown of Fists of Fury by ${$m2/1000} sec.; When Fists of Fury ends, the critical strike chance of Rising Sun Kick is increased by $337482m1% for $337482d.
     xuens_bond                     = { 101208, 392986, 1 }, -- Abilities cast by you or your Storm, Earth, and Fire clones that activate Combo Strikes reduce the cooldown of Invoke Xuen, the White Tiger by ${$s2/-1000}.2 sec, and Xuen's damage is increased by ${$s1}%.
@@ -177,37 +177,37 @@
 spec:RegisterAuras( {
     -- Damage received from $@auracaster increased by $w1%.
     acclamation = {
         id = 451433,
         duration = 12.0,
         max_stack = 1,
     },
     -- Agility increased by $w1%.
     against_all_odds = {
         id = 451061,
-        duration = 6.0,
+        duration = 5.0,
         max_stack = 1,
     },
-    -- The $?c2[damage of your next Rising Sun Kick is increased by $s1% or the healing of your next Vivify is increased by by $s2%]?c3[damage of your next Rising Sun Kick is increased by $s1%][].
+    -- The $?c2[damage of your next Rising Sun Kick is increased by $s1% or the healing of your next Vivify is increased by by $s2%][damage of your next Rising Sun Kick is increased by $s1%].
     august_dynasty = {
         id = 442850,
         duration = 12.0,
         max_stack = 1,
     },
     -- Your next Blackout Kick costs no Chi$?a451495[ and deals $w3% more damage][].
     blackout_kick = {
         id = 116768,
         duration = 15.0,
         max_stack = 1,
 
         -- Affected by:
-        -- courageous_impulse[451495] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.6, 'points': 125.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
+        -- courageous_impulse[451495] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.60000002384, 'points': 125.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
     },
     -- Reduces damage by $w1%.
     bounce_back = {
         id = 390239,
         duration = 4.0,
         max_stack = 1,
     },
     -- Reduces all damage taken by $w1%.
     calming_presence = {
         id = 388664,
@@ -221,21 +221,21 @@
         max_stack = 1,
 
         -- Affected by:
         -- chijis_swiftness[443566] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 75.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_5_VALUE, }
         -- jade_sanctuary[443059] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- jade_sanctuary[443059] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
     },
     -- Chi Burst can be cast!
     chi_burst = {
         id = 460490,
-        duration = 20.0,
+        duration = 30.0,
         max_stack = 1,
     },
     -- Increases the damage done by your next Chi Explosion by $s1%.; Chi Explosion is triggered whenever you use Spinning Crane Kick.
     chi_energy = {
         id = 393057,
         duration = 45.0,
         max_stack = 1,
     },
     -- Healing taken increased by $s1%.
     chi_harmony = {
@@ -259,31 +259,30 @@
     clash = {
         id = 128846,
         duration = 4.0,
         max_stack = 1,
     },
     -- Taking $w1 damage every $t1 sec.
     crackling_jade_lightning = {
         id = 117952,
         duration = 4.0,
         tick_time = 1.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- power_of_the_thunder_king[459809] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 4.0, 'target': TARGET_UNIT_CASTER, 'modifies': CHAINED_TARGETS, }
         -- power_of_the_thunder_king[459809] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- power_of_the_thunder_king[459809] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- the_emperors_capacitor[393039] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 200.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- the_emperors_capacitor[393039] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- hit_combo[196741] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
@@ -294,40 +293,45 @@
     -- Haste increased by $w1%.
     crane_stance = {
         id = 443572,
         duration = 3600,
         max_stack = 1,
     },
     -- Movement slowed by $w1%.
     crashing_momentum = {
         id = 450342,
         duration = 5.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- winds_reach[450514] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
         -- winds_reach[450514] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
         -- winds_reach[450514] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 3000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
     -- Your next Spinning Crane Kick is free and deals an additional $325201s1% damage.
     dance_of_chiji = {
         id = 325202,
         duration = 15.0,
         max_stack = 1,
     },
     -- Your dodge chance is increased by $w1% until you dodge an attack.
     dance_of_the_wind = {
         id = 432180,
         duration = 10.0,
         max_stack = 1,
     },
+    -- Tiger Palm's global cooldown is reduced by $w1%.
+    darting_hurricane = {
+        id = 459841,
+        duration = 30.0,
+        max_stack = 1,
+    },
     -- Spell damage taken reduced by $m1%.
     diffuse_magic = {
         id = 122783,
         duration = 6.0,
         max_stack = 1,
     },
     -- Rooted for $d.
     disable = {
         id = 116706,
         duration = 8.0,
@@ -565,30 +569,30 @@
         duration = 3.0,
         max_stack = 1,
 
         -- Affected by:
         -- hasty_provocation[328670] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 50.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
     },
     -- Restores $w1 health every $t1 sec.$?a448392[; Healing received from $@auracaster increased by $457013s1% for the first $457013d.][]$?e1[; Healing received from $@auracaster increased by $w2%.][]
     renewing_mist = {
         id = 119611,
         duration = 20.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- renewing_mist[119611] #1: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- chi_proficiency[450426] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- chi_proficiency[450426] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- save_them_all[390105] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- save_them_all[390105] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- jadefire_brand[395413] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ALLY, }
-        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.3, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.5, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
     },
     -- Movement speed increased by $w1%.; Immune to movement speed reduction effects.
     ride_the_wind = {
         id = 201447,
         duration = 1.0,
         max_stack = 1,
     },
     -- Nearby enemies will be knocked out of the Ring of Peace.
     ring_of_peace = {
         id = 116844,
@@ -608,25 +612,24 @@
     -- Clash can be reactivated.
     rodeo = {
         id = 355918,
         duration = 3.0,
         max_stack = 1,
     },
     -- Dealing physical damage to nearby enemies every $116847t1 sec.
     rushing_jade_wind = {
         id = 116847,
         duration = 6.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
-        -- windwalker_monk[137025] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 12.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 46.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- windwalker_monk[137025] #26: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 7000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- monk[137022] #0: { 'type': APPLY_AURA, 'subtype': MOD_COOLDOWN_BY_HASTE_REGEN, 'target': TARGET_UNIT_CASTER, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- yulons_knowledge[443625] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 4000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
@@ -639,48 +642,48 @@
     -- Disoriented.
     song_of_chiji = {
         id = 198909,
         duration = 20.0,
         max_stack = 1,
     },
     -- Healing for $w1 every $t1 sec.$?a388681[; Damage taken reduced by $w2%.][]
     soothing_mist = {
         id = 115175,
         duration = 8.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- renewing_mist[119611] #1: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- chi_proficiency[450426] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- chi_proficiency[450426] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- elusive_mists[388681] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -6.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- save_them_all[390105] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- save_them_all[390105] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- jadefire_brand[395413] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ALLY, }
-        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.3, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.5, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
     },
     -- Attacking all nearby enemies for Physical damage every $101546t1 sec.; Movement speed reduced by $s2%.
     spinning_crane_kick = {
         id = 107270,
         duration = 0.0,
         max_stack = 1,
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- windwalker_monk[137025] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 165.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #7: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 231.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- fast_feet[388809] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- crane_vortex[388848] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- crane_vortex[388848] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 30.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- crane_vortex[388848] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
-        -- efficient_training[450989] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- efficient_training[450989] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- temple_training[442743] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- cyclone_strikes[220358] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 18.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
@@ -709,21 +712,21 @@
     },
     -- Movement speed reduced by $s2%.
     strike_of_the_windlord = {
         id = 392983,
         duration = 6.0,
         max_stack = 1,
 
         -- Affected by:
         -- windwalker_monk[137025] #13: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 26.5, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- communion_with_wind[451576] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -10000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- communion_with_wind[451576] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 80.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- communion_with_wind[451576] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
     -- Your next Blackout Kick strikes an additional $m1 $Ltime:times;$?s210802[ and restores ${$210803m~1*$m1/100}.2% mana][].
     teachings_of_the_monastery = {
         id = 202090,
         duration = 20.0,
@@ -751,21 +754,20 @@
         max_stack = 1,
 
         -- Affected by:
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
     },
     -- $w1 Nature damage every $t1 sec.
     touch_of_karma = {
         id = 124280,
         duration = 6.0,
         tick_time = 1.0,
-        pandemic = true,
         max_stack = 1,
     },
     -- Your spirit is tethered to another player, allowing you to use Transcendence: Transfer to teleport to their location.
     transcendence = {
         id = 434767,
         duration = 3600.0,
         max_stack = 1,
     },
     -- Damage of your next Fists of Fury increased by $s1%.
     transfer_the_power = {
@@ -775,21 +777,21 @@
     },
     -- Damage of your next set of Flurry Strikes increased by $w1%.
     vigilant_watch = {
         id = 451233,
         duration = 30.0,
         max_stack = 1,
     },
     -- Your next Vivify is instant and its healing is increased by $s2%.$?c1[; The cost of Vivify is reduced by $s3%.]?c3[; The cost of Vivify is reduced by $s4%.][]
     vivacious_vivification = {
         id = 392883,
-        duration = 3600,
+        duration = 10.0,
         max_stack = 1,
 
         -- Affected by:
         -- windwalker_monk[137025] #15: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
     },
     -- Parry and Avoidance increased by $w1%.
     whirling_steel = {
         id = 451214,
         duration = 6.0,
         max_stack = 1,
@@ -816,55 +818,55 @@
         cast = 0.0,
         cooldown = 3.0,
         gcd = "global",
 
         spend = 3,
         spendType = 'chi',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.847, 'pvp_multiplier': 1.3, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- #1: { 'type': DUMMY, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.77, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.84700000286, 'pvp_multiplier': 1.29999995232, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #1: { 'type': DUMMY, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.76999998093, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, }
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- windwalker_monk[137025] #8: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
-        -- windwalker_monk[137025] #23: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #23: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 32.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- blackout_kick[261916] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -2.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- brawlers_intensity[451485] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_DEST_DB, 'modifies': DAMAGE_HEALING, }
-        -- hardened_soles[391383] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': CRIT_CHANCE, }
-        -- hardened_soles[391383] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
+        -- brawlers_intensity[451485] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 12.0, 'target': TARGET_DEST_DB, 'modifies': DAMAGE_HEALING, }
+        -- hardened_soles[391383] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': CRIT_CHANCE, }
+        -- hardened_soles[391383] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 18.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
         -- shadowboxing_treads[392982] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': CHAINED_TARGETS, }
         -- shadowboxing_treads[392982] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #4: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- vigilant_watch[450993] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blackout_kick[116768] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- blackout_kick[116768] #1: { 'type': APPLY_AURA, 'subtype': ABILITY_IGNORE_AURASTATE, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
-        -- blackout_kick[116768] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- blackout_kick[116768] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
-    -- $?c2[The August Celestials empower you, causing you to radiate ${$443039s1*$s7} healing onto up to $s3 injured allies and ${$443038s1*$s7} Nature damage onto enemies within $s6 yds over $d, split evenly among them. Healing and damage increased by $s1% per target, up to ${$s1*$s3}%.]?c3[The August Celestials empower you, causing you to radiate ${$443038s1*$s7} Nature damage onto enemies and ${$443039s1*$s7} healing onto up to $s3 injured allies within $443038A2 yds over $d, split evenly among them. Healing and damage increased by $s1% per enemy struck, up to ${$s1*$s3}%.][]; You may move while channeling, but casting other healing or damaging spells cancels this effect.; 
+    -- $?c2[The August Celestials empower you, causing you to radiate ${$443039s1*$s7} healing onto up to $s3 injured allies and ${$443038s1*$s7} Nature damage onto enemies within $s6 yds over $d, split evenly among them. Healing and damage increased by $s1% per target, up to ${$s1*$s3}%.][The August Celestials empower you, causing you to radiate ${$443038s1*$s7} Nature damage onto enemies and ${$443039s1*$s7} healing onto up to $s3 injured allies within $443038A2 yds over $d, split evenly among them. Healing and damage increased by $s1% per enemy struck, up to ${$s1*$s3}%.]; You may move while channeling, but casting other healing or damaging spells cancels this effect.; 
     celestial_conduit = {
         id = 443028,
         cast = 0.0,
         cooldown = 90.0,
         gcd = "global",
 
         spend = 0.050,
         spendType = 'mana',
 
         talent = "celestial_conduit",
@@ -883,21 +885,21 @@
         -- Affected by:
         -- chijis_swiftness[443566] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 75.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_5_VALUE, }
         -- jade_sanctuary[443059] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- jade_sanctuary[443059] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
     },
 
     -- You and the target charge each other, meeting halfway then rooting all targets within $128846A1 yards for $128846d.
     clash = {
         id = 324312,
         cast = 0.0,
-        cooldown = 45.0,
+        cooldown = 60.0,
         gcd = "global",
 
         talent = "clash",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- rodeo[355918] #0: { 'type': APPLY_AURA, 'subtype': OVERRIDE_ACTIONBAR_SPELLS_TRIGGERED, 'attributes': ['Suppress Points Stacking'], 'trigger_spell': 324312, 'triggers': clash, 'points': 355919.0, 'value': 324312, 'schools': ['nature', 'frost', 'arcane'], 'target': TARGET_UNIT_CASTER, }
@@ -922,52 +924,51 @@
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 8.0, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
-        -- memory_of_the_monastery[454969] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
+        -- memory_of_the_monastery[454969] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 25.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
     },
 
     -- Channel Jade lightning, causing $o1 Nature damage over $117952d to the target$?a154436[, generating 1 Chi each time it deals damage,][] and sometimes knocking back melee attackers.
     crackling_jade_lightning = {
         id = 117952,
-        cast = 4.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 20,
         spendType = 'energy',
 
         spend = 20,
         spendType = 'energy',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DAMAGE, 'tick_time': 1.0, 'ap_bonus': 0.056, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DAMAGE, 'tick_time': 1.0, 'ap_bonus': 0.05600000173, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 200.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 200.0, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- power_of_the_thunder_king[459809] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 4.0, 'target': TARGET_UNIT_CASTER, 'modifies': CHAINED_TARGETS, }
         -- power_of_the_thunder_king[459809] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- power_of_the_thunder_king[459809] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- the_emperors_capacitor[393039] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 200.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- the_emperors_capacitor[393039] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- hit_combo[196741] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
@@ -1048,52 +1049,51 @@
 
         spend = 0.014,
         spendType = 'mana',
 
         spend = 15,
         spendType = 'energy',
 
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 1.2, 'variance': 0.05, 'target': TARGET_UNIT_CASTER, }
+        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 1.20000004768, 'variance': 0.05000000075, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- windwalker_monk[137025] #21: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 75.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #28: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- profound_rebuttal[392910] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 50.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
         -- vigorous_expulsion[392900] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- vigorous_expulsion[392900] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': CRIT_CHANCE, }
         -- reverse_harm[342928] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 30.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
 
     -- Pummels all targets in front of you, dealing ${5*$117418s1} Physical damage to your primary target and ${5*$117418s1*$s6/100} damage to all other enemies over $113656d. Deals reduced damage beyond $s1 targets. Can be channeled while moving.
     fists_of_fury = {
         id = 113656,
-        cast = 4.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 24.0,
         gcd = "global",
 
         spend = 3,
         spendType = 'chi',
 
         talent = "fists_of_fury",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': PERIODIC_DUMMY, 'tick_time': 0.166, 'trigger_spell': 120086, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': APPLY_AURA, 'subtype': PERIODIC_DUMMY, 'tick_time': 0.166, 'trigger_spell': 117418, 'triggers': fists_of_fury, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
         -- #3: { 'type': APPLY_AURA, 'subtype': MOD_PARRY_PERCENT, 'target': TARGET_UNIT_CASTER, }
-        -- #4: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 1.2075, 'pvp_multiplier': 1.15, 'variance': 0.05, }
+        -- #4: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 1.20749998093, 'pvp_multiplier': 1.14999997616, 'variance': 0.05000000075, }
         -- #5: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- monk[137022] #0: { 'type': APPLY_AURA, 'subtype': MOD_COOLDOWN_BY_HASTE_REGEN, 'target': TARGET_UNIT_CASTER, }
         -- turbo_fists[287681] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 200.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
@@ -1312,33 +1312,32 @@
         spend = 0.025,
         spendType = 'mana',
 
         talent = "rising_sun_kick",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 185099, 'points': 1.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
-        -- windwalker_monk[137025] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 16.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
-        -- windwalker_monk[137025] #14: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #14: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 61.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- monk[137022] #0: { 'type': APPLY_AURA, 'subtype': MOD_COOLDOWN_BY_HASTE_REGEN, 'target': TARGET_UNIT_CASTER, }
         -- fast_feet[388809] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 70.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- brawlers_intensity[451485] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- rising_star[388849] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- rising_star[388849] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- rising_star[388849] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- august_dynasty[442850] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 30.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- august_dynasty[442850] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.67000001669, 'points': 30.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- pressure_point[337482] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': CRIT_CHANCE, }
     },
 
     -- Roll a short distance.
     roll = {
         id = 109132,
         cast = 0.0,
         cooldown = 0.8,
         gcd = "none",
@@ -1362,21 +1361,21 @@
         spend = 1,
         spendType = 'chi',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_TRIGGER_SPELL, 'tick_time': 0.75, 'trigger_spell': 148187, 'points': 5.0, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, }
 
         -- Affected by:
-        -- windwalker_monk[137025] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 12.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 46.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- windwalker_monk[137025] #26: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 7000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- monk[137022] #0: { 'type': APPLY_AURA, 'subtype': MOD_COOLDOWN_BY_HASTE_REGEN, 'target': TARGET_UNIT_CASTER, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- yulons_knowledge[443625] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 4000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
@@ -1409,89 +1408,86 @@
         -- #0: { 'type': CREATE_AREATRIGGER, 'subtype': NONE, 'attributes': ['Position is facing relative'], 'value': 5484, 'schools': ['fire', 'nature', 'shadow', 'arcane'], 'target': TARGET_DEST_CASTER, }
 
         -- Affected by:
         -- peace_and_prosperity[450448] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
     },
 
     -- Heals the target for $o1 over $d.; While channeling, Effuse and Enveloping Mist are instant cast, and will heal the soothed target without breaking the Soothing Mist channel.
     soothing_mist = {
         id = 209525,
         color = 'pvp_talent',
-        cast = 20.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 1.0,
         gcd = "global",
 
         spendType = 'mana',
 
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'amplitude': 1.0, 'tick_time': 0.5, 'sp_bonus': 0.225, 'chain_amp': 100.0, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'amplitude': 1.0, 'tick_time': 0.5, 'sp_bonus': 0.22499999404, 'chain_amp': 100.0, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': CAST_WHILE_WALKING, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- elusive_mists[388681] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -6.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
     },
 
     -- Heals the target for $o1 over $d. While channeling, Enveloping Mist$?s227344[, Surging Mist,][]$?s124081[, Zen Pulse,][] and Vivify may be cast instantly on the target.$?s117907[; Each heal has a chance to cause a Gust of Mists on the target.][]$?s388477[; Soothing Mist heals a second injured ally within $388478A2 yds for $388477s1% of the amount healed.][]
     soothing_mist_115175 = {
         id = 115175,
-        cast = 8.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spendType = 'mana',
 
         spendType = 'energy',
 
         spendType = 'energy',
 
         talent = "soothing_mist_115175",
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'amplitude': 1.0, 'tick_time': 1.0, 'sp_bonus': 1.4, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'amplitude': 1.0, 'tick_time': 1.0, 'sp_bonus': 1.39999997616, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_DAMAGE_PERCENT_TAKEN, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_TARGET_ALLY, }
 
         -- Affected by:
         -- renewing_mist[119611] #1: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- chi_proficiency[450426] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- chi_proficiency[450426] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- elusive_mists[388681] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -6.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- save_them_all[390105] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- save_them_all[390105] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- jadefire_brand[395413] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ALLY, }
-        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.3, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.5, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
         from = "class_talent",
     },
 
     -- Jabs the target in the throat, interrupting spellcasting and preventing any spell from that school of magic from being cast for $d.
     spear_hand_strike = {
         id = 116705,
         cast = 0.0,
         cooldown = 15.0,
         gcd = "none",
 
         talent = "spear_hand_strike",
         startsCombat = true,
-        interrupt = true,
 
         -- Effects:
         -- #0: { 'type': INTERRUPT_CAST, 'subtype': NONE, 'mechanic': interrupted, 'points': 1.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- Spin while kicking in the air, dealing $?s137025[${4*$107270s1*$<CAP>/$AP}][${4*$107270s1}] Physical damage over $d to all enemies within $107270A1 yds. Deals reduced damage beyond $s1 targets.$?a220357[; Spinning Crane Kick's damage is increased by $220358s1% for each unique target you've struck in the last $220358d with Tiger Palm, Blackout Kick, or Rising Sun Kick. Stacks up to $228287i times.][]
     spinning_crane_kick = {
         id = 101546,
-        cast = 1.5,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 2,
         spendType = 'chi',
 
         spend = 0.010,
         spendType = 'mana',
 
         spend = 40,
@@ -1500,24 +1496,24 @@
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_TRIGGER_SPELL, 'tick_time': 0.5, 'trigger_spell': 107270, 'points': 5.0, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': APPLY_AURA, 'subtype': DUMMY, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 15.0, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- fast_feet[388809] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- crane_vortex[388848] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- crane_vortex[388848] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 30.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- crane_vortex[388848] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
-        -- efficient_training[450989] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- dance_of_chiji[325202] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -2.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
+        -- efficient_training[450989] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- dance_of_chiji[325202] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'points': -2.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
     -- Split into 3 elemental spirits for $d, each spirit dealing ${100+$m1}% of normal damage and healing.; You directly control the Storm spirit, while Earth and Fire spirits mimic your attacks on nearby enemies.; While active, casting Storm, Earth, and Fire again will cause the spirits to fixate on your target.
     storm_earth_and_fire = {
         id = 137639,
         cast = 0.0,
         cooldown = 16.0,
         gcd = "none",
 
@@ -1554,70 +1550,70 @@
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_DECREASE_SPEED, 'mechanic': snared, 'points': -50.0, 'radius': 12.0, 'target': TARGET_UNIT_RECT_CASTER_ENEMY, }
         -- #2: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 395519, 'target': TARGET_UNIT_CASTER, }
         -- #3: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 395521, 'value': 500, 'schools': ['fire', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- windwalker_monk[137025] #13: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 26.5, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- communion_with_wind[451576] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -10000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- communion_with_wind[451576] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 80.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- communion_with_wind[451576] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- ordered_elements[451462] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
     -- Strike with the palm of your hand, dealing $s1 Physical damage.$?a137384[; Tiger Palm has an $137384m1% chance to make your next Blackout Kick cost no Chi.][]$?a137023[; Reduces the remaining cooldown on your Brews by $s3 sec.][]$?a137025[; Generates $s2 Chi.][]
     tiger_palm = {
         id = 100780,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 50,
         spendType = 'energy',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.27027, 'pvp_multiplier': 1.3, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.27026998997, 'pvp_multiplier': 1.29999995232, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': ENERGIZE, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, 'resource': chi, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- windwalker_monk[137025] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- windwalker_monk[137025] #3: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
-        -- windwalker_monk[137025] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 16.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- windwalker_monk[137025] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 125.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 159.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #11: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -500.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
         -- windwalker_monk[137025] #13: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 26.5, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk[137025] #27: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- windwalker_monk_twohand_adjustment[346104] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- windwalker_monk_twohand_adjustment[346104] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- efficient_training[450989] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- efficient_training[450989] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- inner_peace[397768] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- storm_earth_and_fire[137639] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- storm_earth_and_fire[137639] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -60.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- touch_of_the_tiger[388856] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- touch_of_the_tiger[388856] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- xuens_guidance[442687] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- tigereye_brew[247483] #0: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'value': 8, 'schools': ['nature'], 'target': TARGET_UNIT_CASTER, }
         -- tigereye_brew[247483] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- fatal_touch[450832] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- darting_hurricane[459841] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
-        -- martial_mixture[451457] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- darting_hurricane[459841] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': GLOBAL_COOLDOWN, }
+        -- darting_hurricane[459841] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'attributes': ['Suppress Points Stacking'], 'points': 1000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
+        -- martial_mixture[451457] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.40000000596, 'points': 8.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- hit_combo[196741] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
 
     -- Consumes up to $s2 stacks of Tigereye Brew to empower your Physical abilities with wind for $d per stack consumed. Damage of your strikes are reduced, but bypass armor.; For each $s3 Chi you consume, you gain a stack of Tigereye Brew.
     tigereye_brew = {
         id = 247483,
         color = 'pvp_talent',
         cast = 0.0,
         cooldown = 1.0,
@@ -1658,21 +1654,21 @@
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- fatal_touch[394123] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -90000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- meridian_strikes[391330] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.4, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- meridian_strikes[391330] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.40000000596, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
 
     -- Instantly kills any creature under level $m2, or player under $m3% health.
     touch_of_fatality = {
         id = 169340,
         cast = 0.0,
         cooldown = 90.0,
         gcd = "global",
 
         startsCombat = true,
@@ -1688,21 +1684,21 @@
         id = 122470,
         cast = 0.0,
         cooldown = 90.0,
         gcd = "none",
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': DUMMY, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': SCHOOL_ABSORB, 'value': 127, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'value1': -30, 'target': TARGET_UNIT_CASTER, }
-        -- #2: { 'type': DUMMY, 'subtype': NONE, 'pvp_multiplier': 1.6, }
+        -- #2: { 'type': DUMMY, 'subtype': NONE, 'pvp_multiplier': 1.60000002384, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, }
         -- #4: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Split your body and spirit, leaving your spirit behind for $d. Use Transcendence: Transfer to swap locations with your spirit.
     transcendence = {
         id = 101643,
         cast = 0.0,
         cooldown = 10.0,
         gcd = "global",
@@ -1774,38 +1770,39 @@
 
         spend = 30,
         spendType = 'energy',
 
         spend = 30,
         spendType = 'energy',
 
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 2.58, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- #0: { 'type': HEAL, 'subtype': NONE, 'sp_bonus': 2.57999992371, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ALLY, }
 
         -- Affected by:
-        -- windwalker_monk[137025] #19: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.34, 'points': 150.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- windwalker_monk[137025] #19: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.34000000358, 'points': 150.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- renewing_mist[119611] #1: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'target': TARGET_UNIT_TARGET_ALLY, }
         -- chi_proficiency[450426] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- temple_training[442743] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- temple_training[442743] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- chi_proficiency[450426] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- temple_training[442743] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- temple_training[442743] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- vivacious_vivification[392883] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -1500.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- vivacious_vivification[392883] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- vivacious_vivification[392883] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -75.0, 'target': TARGET_UNIT_CASTER, 'modifies': IGNORE_SHAPESHIFT, }
         -- vivacious_vivification[392883] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -75.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- august_dynasty[442850] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 50.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- escape_from_reality[343249] #2: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'points': 70.0, 'target': TARGET_UNIT_CASTER, }
         -- save_them_all[390105] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- save_them_all[390105] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- jadefire_brand[395413] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ALLY, }
-        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.3, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
+        -- chi_harmony[423439] #0: { 'type': APPLY_AURA, 'subtype': MOD_HEALING_RECEIVED, 'pvp_multiplier': 0.5, 'points': 50.0, 'target': TARGET_UNIT_TARGET_ALLY, }
     },
 
     -- Performs a devastating whirling upward strike, dealing ${3*$158221s1} damage to all nearby enemies and an additional $451767s1 damage to the first target struck. Damage reduced beyond $s1 targets.; Only usable while both Fists of Fury and Rising Sun Kick are on cooldown.
     whirling_dragon_punch = {
         id = 152175,
         cast = 0.0,
         cooldown = 30.0,
         gcd = "global",
 
         talent = "whirling_dragon_punch",
