--- "D:\\Development\\WoW Python Tools\\Output\\11.0.2.55763\\WarlockDemonology.lua"	2024-07-21 09:48:32.495915800 -1000
+++ "D:\\Development\\WoW Python Tools\\Output\\11.0.5.56865\\WarlockDemonology.lua"	2024-10-21 14:01:56.351073500 -1000
@@ -1,12 +1,12 @@
 -- WarlockDemonology.lua
--- July 2024
+-- October 2024
 
 local addon, ns = ...
 local Hekili = _G[ addon ]
 local class, state = Hekili.Class, Hekili.State
 
 local spec = Hekili:NewSpecialization( 266 )
 
 -- Resources
 spec:RegisterResource( Enum.PowerType.Mana )
 spec:RegisterResource( Enum.PowerType.SoulShards )
@@ -68,21 +68,21 @@
     call_dreadstalkers             = { 101894, 104316, 1 }, -- Summons $s1 ferocious Dreadstalkers to attack the target for $193332d.
     carnivorous_stalkers           = { 101887, 386194, 1 }, -- Your Dreadstalkers' attacks have a $s1% chance to trigger an additional Dreadbite.
     cloven_souls                   = { 94849, 428517, 1 }, -- Enemies damaged by your Overlord have their souls cloven, increasing damage taken by you and your pets by $434424s1% for $434424d.
     cruelty_of_kerxan              = { 94848, 429902, 1 }, -- $?s137044[Summon Demonic Tyrant][Summon Infernal] grants Diabolic Ritual and reduces its duration by ${$s1/1000} sec.
     demoniac                       = { 101891, 426115, 1 }, -- [264178] Send the fiery soul of a fallen demon at the enemy, causing $s1 Shadowflame damage.$?c2[; Generates 2 Soul Shards.][]
     demoniacs_fervor               = { 94832, 449629, 1 }, -- Your demonic soul deals $s1% increased damage to $?s137043[targets affected by your Unstable Affliction.][the main target of Hand of Gul'dan.]
     demonic_brutality              = { 101920, 453908, 1 }, -- Critical strikes from your spells and your demons deal $s1% increased damage.
     demonic_calling                = { 101903, 205145, 1 }, -- Shadow Bolt$?s264178[ and Demonbolt have][ has] a $s3% chance to make your next Call Dreadstalkers cost $s1 fewer Soul $LShard:Shards; and have no cast time.
     demonic_soul                   = { 94851, 449614, 1 }, -- A demonic entity now inhabits your soul, allowing you to detect if a Soul Shard has a Succulent Soul when it's generated. ; A Succulent Soul empowers your next $?s137043[Malefic Rapture, increasing its damage by $449793s2%, and unleashing your demonic soul to deal an additional $449801s1 Shadow damage.][Hand of Gul'dan, increasing its damage by $449793s3%, and unleashing your demonic soul to deal an additional $449801s1 Shadow damage.]
     demonic_strength               = { 101890, 267171, 1 }, -- Infuse your Felguard with demonic strength and command it to charge your target and unleash a Felstorm that will deal $s2% increased damage.
-    diabolic_ritual                = { 94855, 428514, 1 }, -- Spending a Soul Shard on a damaging spell grants Diabolic Ritual for $431944d. While Diabolic Ritual is active, each Soul Shard spent on a damaging spell reduces its duration by $s1 sec.; When Diabolic Ritual expires you gain Demonic Art, causing your next $?s137044[Hand of Gul'dan][Chaos Bolt, Rain of Fire, or Shadowburn] to summon an Overlord, Mother of Chaos, or Pit Lord that unleashes a devastating attack against your enemies.
+    diabolic_ritual                = { 94855, 428514, 1 }, -- $?s137044[Spending a Soul Shard on a damaging spell grants Diabolic Ritual for $431944d. While Diabolic Ritual is active, each Soul Shard spent on a damaging spell reduces its duration by $s1 sec.][Casting Chaos Bolt, Rain of Fire, or Shadowburn grants Diabolic Ritual for $431944d. If Diabolic Ritual is already active, its duration is reduced by $s2 sec instead.]; When Diabolic Ritual expires you gain Demonic Art, causing your next $?s137044[Hand of Gul'dan][Chaos Bolt, Rain of Fire, or Shadowburn] to summon an Overlord, Mother of Chaos, or Pit Lord that unleashes a devastating attack against your enemies.
     doom                           = { 101919, 460551, 1 }, -- When Demonbolt consumes a Demonic Core it inflicts impending doom upon the target, dealing $460555s1 Shadow damage to enemies within $460555a1 yds of its target after $460553d or when removed. Damage is reduced beyond $s2 targets.; Consuming a Demonic Core reduces the duration of Doom by ${$s1/1000} sec.
     doom_eternal                   = { 101906, 455585, 1 }, -- Demonic Cores reduce the duration of Doom by an additional ${$s1/1000} sec.
     dread_calling                  = { 101889, 387391, 1 }, -- Each Soul Shard spent on Hand of Gul'dan increases the damage of your next Call Dreadstalkers by $s1%.
     dreadlash                      = { 101888, 264078, 1 }, -- When your Dreadstalkers charge into battle, their Dreadbite attack now hits all targets within $271971A1 yards and deals $s1% more damage.
     eternal_servitude              = { 94824, 449707, 1 }, -- Fel Domination cooldown is reduced by ${$s1/-1000} sec.
     feast_of_souls                 = { 94823, 449706, 1 }, -- When you kill a target, you have a chance to generate a Soul Shard that is guaranteed to be a Succulent Soul.
     fel_invocation                 = { 101897, 428351, 1 }, -- Soul Strike deals $s1% increased damage and generates a Soul Shard.
     fel_sunder                     = { 101911, 387399, 1 }, -- Each time Felstorm deals damage, it increases the damage the target takes from you and your pets by $s1% for $387402d, up to ${$s1*$387402u}%.
     fiendish_oblation              = { 101912, 455569, 1 }, -- Damage dealt by Grimoire: Felguard is increased by an additional $s1% and you gain a Demonic Core when Grimoire: Felguard ends.
     flames_of_xoroth               = { 94833, 429657, 1 }, -- Fire damage increased by $s1% and damage dealt by your demons is increased by $s3%.
@@ -122,21 +122,21 @@
     shadowtouched                  = { 101910, 453619, 1 }, -- Wicked Maw causes the target to take $s1% additional Shadow damage from your demons.
     shared_fate                    = { 94823, 449704, 1 }, -- When you kill a target, its tortured soul is flung into a nearby enemy for $450591d. This effect inflicts $450593s1 Shadow damage to enemies within $450593a1 yds every $450591t1 sec.; Deals reduced damage beyond $s1 targets.
     soul_anathema                  = { 94847, 449624, 1 }, -- Unleashing your demonic soul bestows a fiendish entity unto the soul of its targets, dealing $450538o1 Shadow damage over $450538d.; If this effect is reapplied, any remaining damage will be added to the new Soul Anathema.
     soul_strike                    = { 101899, 428344, 1 }, -- [267964] Strike into the soul of the enemy, dealing $<damage> Shadow damage.$?s428351[; Generates 1 Soul Shard.][]
     souletched_circles             = { 94836, 428911, 1 }, -- You always gain the benefit of Soulburn when casting Demonic Circle: Teleport, increasing your movement speed by 50% and making you immune to snares and roots for 6 sec.
     spiteful_reconstitution        = { 101901, 428394, 1 }, -- Implosion deals $s1% increased damage. Consuming a Demonic Core has a chance to summon a Wild Imp.
     summon_demonic_tyrant          = { 101905, 265187, 1 }, -- Summon a Demonic Tyrant to increase the duration of your Dreadstalkers, Vilefiend, Felguard, and up to $s3 of your Wild Imps by ${$265273m3/1000} sec. Your Demonic Tyrant increases the damage of affected demons by $265273s1%, while damaging your target.$?s334585[; Generates ${$s2/10} Soul Shards.][]
     summon_vilefiend               = { 101900, 264119, 1 }, -- Summon a Vilefiend to fight for you for the next $d.
     the_expendables                = { 101902, 387600, 1 }, -- When your Wild Imps expire or die, your other demons are inspired and gain 1% additional damage, stacking up to $387601u times.
     the_houndmasters_gambit        = { 101898, 455572, 1 }, -- Your Dreadstalkers deal $455611s1% increased damage while your Vilefiend is active.
-    touch_of_rancora               = { 94856, 429893, 1 }, -- Demonic Art increases the damage of your next $?s137044[Hand of Gul'dan][Chaos Bolt, Rain of Fire, or Shadowburn] by $s1% and reduces its cast time by $s2%.
+    touch_of_rancora               = { 94856, 429893, 1 }, -- Demonic Art increases the damage of your next $?s137044[Hand of Gul'dan][Chaos Bolt, Rain of Fire, or Shadowburn] by $s1% and reduces its cast time by $s2%.; $?s137044[][Casting Chaos Bolt reduces the duration of Diabolic Ritual by $s3 additional sec.]
     umbral_blaze                   = { 101915, 405798, 1 }, -- Hand of Gul'dan has a $s1% chance to burn its target for ${$405802s1*(1+$s2/100)} additional Shadowflame damage every $405802t1 sec for $405802d.; If this effect is reapplied, any remaining damage will be added to the new Umbral Blaze.
     wicked_maw                     = { 101926, 267170, 1 }, -- Dreadbite causes the target to take $270569s1% additional Shadowflame damage from your spell and abilities for the next $270569d.
     wicked_reaping                 = { 94821, 449631, 1 }, -- Damage dealt by your demonic soul is increased by $s1%.; Consuming $?s137043[Nightfall][Demonic Core] feeds the demonic entity within you, causing it to appear and deal $?s137043[$449826s1][${$449826s1*($s2/100)}] Shadow damage to your target.
 } )
 
 -- PvP Talents
 spec:RegisterPvpTalents( {
     bonds_of_fel     = 5545, -- (353753) Encircle enemy players with Bonds of Fel. If any affected player leaves the $a2 yd radius they explode, dealing $353813s1 Fire damage split amongst all nearby enemies.
     call_fel_lord    = 162 , -- (212459) Summon a fel lord to guard the location for $d. Any enemy that comes within $213688A1 yards will suffer $<FelCleave> Physical damage, and players struck will be stunned for $213688d.
     call_observer    = 165 , -- (201996) Summons a demonic Observer to keep a watchful eye over the area for $d.; Anytime an enemy within $m2 yards casts a harmful magical spell, the Observer will deal up to $212529s2% of the target's maximum health in Shadow damage.
@@ -186,48 +186,44 @@
     -- Invulnerable, but unable to act.
     banish = {
         id = 710,
         duration = 30.0,
         max_stack = 1,
     },
     -- Movement speed increased by $s1%.
     burning_rush = {
         id = 111400,
         duration = 3600,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- fiendish_stride[386110] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- fiendish_stride[386110] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
     },
     -- Damage taken from you and your pets is increased by $s1%.
     cloven_soul = {
         id = 434424,
         duration = 15.0,
         max_stack = 1,
     },
     -- Suffering $w1 Shadow damage every $t1 sec.
     corruption = {
         id = 146739,
         duration = 14.0,
         tick_time = 2.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- demonology_warlock[137044] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- demonology_warlock[137044] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- demonic_brutality[453908] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 4.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
         -- sataiels_volition[449637] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- absolute_corruption[196103] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- wither[445465] #0: { 'type': APPLY_AURA, 'subtype': OVERRIDE_ACTIONBAR_SPELLS, 'spell': 445468, 'value1': 3, 'target': TARGET_UNIT_CASTER, }
     },
     -- Time between attacks increased by $w1%. $?e1[Chance to critically strike reduced by $w2%.][]
     curse_of_weakness = {
         id = 702,
         duration = 120.0,
         max_stack = 1,
 
         -- Affected by:
         -- amplify_curse[328774] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
     },
@@ -269,29 +265,27 @@
     -- [428524] Your next Soul Shard spent summons an Overlord that unleashes a devastating attack.
     diabolic_ritual_overlord = {
         id = 431944,
         duration = 20.0,
         max_stack = 1,
     },
     -- Suffering $s1 Shadow damage every $t1 seconds.; Restoring health to the Warlock.
     drain_life = {
         id = 234153,
         duration = 5.0,
-        pandemic = true,
         max_stack = 1,
     },
     -- $?j1g[Increases ground speed by $j1g%. ][]$?j1f[Increases flight speed by $j1f%. ][]$?j1s[Increases swim speed by $j1s%. ][]
     dreadsteed = {
         id = 23161,
         duration = 3600,
         tick_time = 0.5,
-        pandemic = true,
         max_stack = 1,
     },
     -- Healing for $m1% of maximum health every $t1 sec.; Spell casts are not delayed by taking damage.
     empowered_healthstone = {
         id = 262080,
         duration = 6.0,
         max_stack = 1,
     },
     -- Controlling Eye of Kilrogg.; Detecting Invisibility.
     eye_of_kilrogg = {
@@ -301,43 +295,41 @@
     },
     -- Disoriented.
     fear = {
         id = 5782,
         duration = 0.0,
         max_stack = 1,
 
         -- Affected by:
         -- fear[342914] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_9, }
     },
-    -- Damage is being delayed every $t1 sec.
+    -- $w1 damage is being delayed every $387846t1 sec.; Damage Remaining: $w2
     fel_armor = {
-        id = 387846,
-        duration = 3600,
-        tick_time = 0.5,
-        pandemic = true,
+        id = 387847,
+        duration = 5.0,
         max_stack = 1,
     },
     -- Stunned.
     fel_cleave = {
         id = 213688,
         duration = 1.0,
         max_stack = 1,
     },
     -- Imp, Voidwalker, Succubus, Felhunter, or Felguard casting time reduced by $/1000;S1 sec.
     fel_domination = {
         id = 333889,
         duration = 15.0,
         max_stack = 1,
 
         -- Affected by:
         -- fel_pact[386113] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -60000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- eternal_servitude[449707] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -90000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
+        -- eternal_servitude[449707] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.33333399892, 'points': -90000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
     },
     -- Damage taken reduced by $w1%.
     fel_resilience = {
         id = 386869,
         duration = 5.0,
         max_stack = 1,
     },
     -- Damage taken from $@auracaster and their pets is increased by $s1%.
     fel_sunder = {
         id = 387402,
@@ -345,49 +337,47 @@
         max_stack = 1,
 
         -- Affected by:
         -- fel_sunder[387399] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
         -- fel_sunder[387399] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
     },
     -- $?j1g[Increases ground speed by $j1g%. ][]$?j1f[Increases flight speed by $j1f%. ][]$?j1s[Increases swim speed by $j1s%. ][]
     felsteed = {
         id = 5784,
         duration = 3600,
-        pandemic = true,
         max_stack = 1,
     },
     -- Summoned by a Grimoire of Service.; Damage done increased by $s1%.
     grimoire_of_service = {
         id = 216187,
         duration = 3600,
         max_stack = 1,
 
         -- Affected by:
         -- fiendish_oblation[455569] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': SPELL_EFFECTIVENESS, }
     },
     -- Transferring health.
     health_funnel = {
         id = 755,
         duration = 5.0,
         tick_time = 1.0,
-        pandemic = true,
         max_stack = 1,
     },
     -- Disoriented.
     howl_of_terror = {
         id = 5484,
         duration = 20.0,
         max_stack = 1,
 
         -- Affected by:
         -- annihilans_bellow[429072] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- annihilans_bellow[429072] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
+        -- annihilans_bellow[429072] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.40000000596, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
     },
     -- Damage done increased by $s2%.
     imp_gang_boss = {
         id = 387458,
         duration = 3600,
         max_stack = 1,
     },
     -- Soul Leech can absorb an additional $s1% of your maximum health.
     infernal_bulwark = {
         id = 434561,
@@ -440,31 +430,32 @@
         max_stack = 1,
 
         -- Affected by:
         -- darkfury[264874] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
         -- darkfury[264874] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
     },
     -- Dealing $450593s1 Shadow damage to enemies within $450593a1 yds every $t1 sec.
     shared_fate = {
         id = 450591,
         duration = 3.0,
+        tick_time = 1.0,
         max_stack = 1,
     },
     -- Dealing $o1 Shadow damage over $d.
     soul_anathema = {
         id = 450538,
         duration = 10.0,
         tick_time = 1.0,
         max_stack = 1,
 
         -- Affected by:
-        -- mastery_master_demonologist[77219] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'sp_bonus': 1.45, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- mastery_master_demonologist[77219] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'sp_bonus': 1.45000004768, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- quietus[449634] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 25.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- quietus[449634] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- quietus[449634] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': -20.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
     -- Absorbs $w1 damage.
     soul_leech = {
         id = 108366,
         duration = 15.0,
         max_stack = 1,
 
@@ -478,21 +469,21 @@
         duration = 8.0,
         max_stack = 1,
     },
     -- Increases the duration of your next Unstable Affliction by ${$m1/1000} sec.
     soulburn = {
         id = 213398,
         duration = 10.0,
         max_stack = 1,
 
         -- Affected by:
-        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45000004768, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- shadow_invocation[422054] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- wicked_maw[270569] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
     -- Movement speed increased by $s1%. Immune to snares and roots.
     soulburn_demonic_circle = {
         id = 387633,
         duration = 6.0,
         max_stack = 1,
     },
     -- Maximum health is increased by $s1%.
@@ -532,21 +523,20 @@
     the_houndmasters_gambit = {
         id = 455611,
         duration = 30.0,
         max_stack = 1,
     },
     -- Dealing $w1 Shadowflame damage every $t1 sec for $d.
     umbral_blaze = {
         id = 405802,
         duration = 6.0,
         tick_time = 2.0,
-        pandemic = true,
         max_stack = 1,
     },
     -- Underwater Breathing. Swim speed increased by $s2%.
     unending_breath = {
         id = 5697,
         duration = 600.0,
         max_stack = 1,
     },
     -- Damage taken reduced by $w3%; Immune to interrupt and silence effects.
     unending_resolve = {
@@ -740,26 +730,25 @@
         gcd = "global",
 
         spend = 0.010,
         spendType = 'mana',
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 146739, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, }
-        -- #2: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 0.138, 'pvp_multiplier': 1.25, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #2: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 0.13799999654, 'pvp_multiplier': 1.25, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- sataiels_volition[449637] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- absolute_corruption[196103] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- wither[445465] #0: { 'type': APPLY_AURA, 'subtype': OVERRIDE_ACTIONBAR_SPELLS, 'spell': 445468, 'value1': 3, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- [386646] When you use a Healthstone, gain $s2% Leech for $386647d.
     create_healthstone = {
         id = 6201,
         cast = 3.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.020,
@@ -820,21 +809,21 @@
         gcd = "none",
 
         talent = "dark_pact",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': SCHOOL_ABSORB, 'value': 127, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, 'sp_bonus': 5.0, }
-        -- #4: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 1.0, 'sp_bonus': 0.182332, 'target': TARGET_UNIT_CASTER, }
+        -- #4: { 'type': APPLY_AURA, 'subtype': PERIODIC_HEAL, 'tick_time': 1.0, 'sp_bonus': 1.64583003521, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- frequent_donor[386686] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
         -- ichor_of_devils[386664] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- ichor_of_devils[386664] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 300.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
     },
 
     -- Creates a demonic gateway between two locations. Activating the gateway transports the user to the other gateway. Each player can use a Demonic Gateway only once per $<GatewayCooldown> sec.
     demonic_gateway = {
         id = 111771,
@@ -884,31 +873,30 @@
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': SCRIPT_EFFECT, 'subtype': NONE, 'points': 24710.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': ENERGIZE, 'subtype': NONE, 'points': 3.0, 'target': TARGET_UNIT_CASTER, 'resource': soul_shards, }
     },
 
     -- [386619] Drain Life heals for $s1% more while below $s2% health.
     drain_life = {
         id = 234153,
-        cast = 5.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spendType = 'mana',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_LEECH, 'amplitude': 5.0, 'tick_time': 1.0, 'sp_bonus': 0.15, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_LEECH, 'amplitude': 5.0, 'tick_time': 1.0, 'sp_bonus': 0.15000000596, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- Summons an Eye of Kilrogg and binds your vision to it. The eye is stealthed and moves quickly but is very fragile.
     eye_of_kilrogg = {
         id = 126,
         cast = 45.0,
         channeled = true,
         cooldown = 0.0,
         gcd = "global",
 
@@ -937,48 +925,49 @@
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'mechanic': fleeing, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- fear[342914] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 100.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_9, }
     },
 
     -- [212459] Summon a fel lord to guard the location for $d. Any enemy that comes within $213688A1 yards will suffer $<FelCleave> Physical damage, and players struck will be stunned for $213688d.
     fel_cleave = {
         id = 213688,
-        cast = 1.3,
+        cast = 1.0,
+        channeled = true,
         cooldown = 2.0,
         gcd = "none",
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 4.68, 'variance': 0.05, 'radius': 6.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 4.67999982834, 'variance': 0.05000000075, 'radius': 6.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_STUN, 'mechanic': stunned, 'points': 1.0, 'radius': 6.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ENEMY, }
     },
 
     -- Your next Imp, Voidwalker, Incubus, Succubus, Felhunter, or Felguard Summon spell is free and has its casting time reduced by $s1%.; 
     fel_domination = {
         id = 333889,
         cast = 0.0,
         cooldown = 180.0,
         gcd = "none",
 
         talent = "fel_domination",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -90.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
 
         -- Affected by:
         -- fel_pact[386113] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -60000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- eternal_servitude[449707] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -90000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
+        -- eternal_servitude[449707] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.33333399892, 'points': -90000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
     },
 
     -- Summons a Felguard who attacks the target for $d that deals $216187s1% increased damage.; This Felguard will stun and interrupt their target when summoned.
     grimoire_felguard = {
         id = 111898,
         color = 'summon',
         cast = 0.0,
         cooldown = 120.0,
         gcd = "global",
 
@@ -1019,29 +1008,28 @@
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'radius': 8.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- Sacrifices ${$m3*$d/$t1}% of your maximum health to heal your summoned Demon for twice as much over $d.
     health_funnel = {
         id = 755,
-        cast = 5.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         startsCombat = false,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DUMMY, 'amplitude': 2.2, 'tick_time': 1.0, 'points': 5.0, 'target': TARGET_UNIT_PET, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_DUMMY, 'amplitude': 2.20000004768, 'tick_time': 1.0, 'points': 5.0, 'target': TARGET_UNIT_PET, }
     },
 
     -- Let loose a terrifying howl, causing $i enemies within $A1 yds to flee in fear, disorienting them for $d. Damage may cancel the effect.
     howl_of_terror = {
         id = 5484,
         cast = 0.0,
         cooldown = 40.0,
         gcd = "global",
 
         spend = 0.018,
@@ -1049,46 +1037,47 @@
 
         talent = "howl_of_terror",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': MOD_FEAR, 'points': 1.0, 'value': 1, 'schools': ['physical'], 'radius': 10.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': USE_NORMAL_MOVEMENT_SPEED, 'points': 4.0, 'radius': 10.0, 'target': TARGET_SRC_CASTER, 'target2': TARGET_UNIT_SRC_AREA_ENEMY, }
 
         -- Affected by:
         -- annihilans_bellow[429072] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- annihilans_bellow[429072] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
+        -- annihilans_bellow[429072] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'pvp_multiplier': 0.40000000596, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': RADIUS, }
     },
 
     -- Demonic forces suck all of your Wild Imps toward the target, and then cause them to violently explode, dealing $196278s2 Shadowflame damage to all enemies within $196278A3 yards.
     implosion = {
         id = 196277,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.020,
         spendType = 'mana',
 
         talent = "implosion",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, 'radius': 100.0, 'target': TARGET_UNIT_CASTER_AND_SUMMONS, }
 
         -- Affected by:
-        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45000004768, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- flames_of_xoroth[429657] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- flames_of_xoroth[429657] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- spiteful_reconstitution[428394] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- wicked_maw[270569] #0: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'points': 20.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- wicked_maw[270569] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- Horrifies an enemy target into fleeing, incapacitating for $6789d and healing you for $108396m1% of maximum health.
     mortal_coil = {
         id = 6789,
         cast = 0.0,
         cooldown = 45.0,
         gcd = "global",
 
         spend = 0.020,
@@ -1118,20 +1107,21 @@
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': REFLECT_SPELLS, 'amplitude': 1.0, 'points': 100.0, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Instantly sacrifice up to $s1 Wild Imps, generating $s1 charges of Demonic Core that cause Demonbolt to deal $334581s1% additional damage.
     power_siphon = {
         id = 264130,
         cast = 0.0,
+        channeled = true,
         cooldown = 30.0,
         gcd = "global",
 
         talent = "power_siphon",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'radius': 40.0, 'target': TARGET_UNIT_CASTER_AND_SUMMONS, }
     },
 
@@ -1145,40 +1135,38 @@
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Begins a ritual that sacrifices a random participant to summon a doomguard. Requires the caster and 4 additional party members to complete the ritual.
     ritual_of_doom = {
         id = 342601,
-        cast = 180.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 3600.0,
         gcd = "global",
 
         spend = 10,
         spendType = 'soul_shards',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'attributes': ["Don't Fail Spell On Targeting Failure"], 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': TRANS_DOOR, 'subtype': NONE, 'points': 1.0, 'value': 177193, 'schools': ['physical', 'nature', 'shadow'], 'radius': 5.0, 'target': TARGET_DEST_CASTER_FRONT, }
     },
 
     -- Begins a ritual to create a summoning portal, requiring the caster and 2 allies to complete. This portal can be used to summon party and raid members.
     ritual_of_summoning = {
         id = 698,
-        cast = 120.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 120.0,
         gcd = "global",
 
         spend = 0.050,
         spendType = 'mana',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': TRANS_DOOR, 'subtype': NONE, 'value': 194108, 'schools': ['fire', 'nature', 'frost', 'shadow'], 'radius': 0.0, 'target': TARGET_DEST_CASTER_FRONT, }
@@ -1191,29 +1179,28 @@
         cast = 2.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 0.015,
         spendType = 'mana',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 0.702, 'pvp_multiplier': 1.9, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'sp_bonus': 0.70200002193, 'pvp_multiplier': 2.47000002861, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- demonology_warlock[137044] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- demonology_warlock[137044] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- demonology_warlock[137044] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 60.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- sargerei_technique[405955] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 8.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- demonic_brutality[453908] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 4.0, 'target': TARGET_UNIT_CASTER, 'modifies': SHOULD_NEVER_SEE_15, }
-        -- necrolyte_teachings[449620] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- necrolyte_teachings[449620] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.5, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- rune_of_shadows[453744] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -25.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- rune_of_shadows[453744] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
 
     -- Conjure a Shadow Rift at the target location lasting $d. Enemy players within the rift when it expires are teleported to your Demonic Circle.; Must be within $s2 yds of your Demonic Circle to cast.
     shadow_rift = {
         id = 353294,
         color = 'pvp_talent',
         cast = 0.0,
         cooldown = 60.0,
@@ -1313,21 +1300,21 @@
 
         spend = 1,
         spendType = 'soul_shards',
 
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
 
         -- Affected by:
-        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- mastery_master_demonologist[77219] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'sp_bonus': 1.45000004768, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- shadow_invocation[422054] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- wicked_maw[270569] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ENEMY, }
         from = "affected_by_mastery",
     },
 
     -- Stores the soul of the target party or raid member, allowing resurrection upon death.$?a231811[ Also castable to resurrect a dead target.][] Targets resurrect with $3026s2% health and at least $3026s1% mana.
     soulstone = {
         id = 20707,
         cast = 3.0,
         cooldown = 600.0,
@@ -1511,21 +1498,22 @@
 
         -- Affected by:
         -- fel_domination[333889] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -90.0, 'target': TARGET_UNIT_CASTER, 'modifies': CAST_TIME, }
         -- fel_domination[333889] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
     },
 
     -- Thal'kiel drains $s2% of the life from your demon servants and unleashes a blast of Shadow damage at your current target equal to the life he stole. 
     thalkiels_consumption = {
         id = 211714,
         color = 'artifact',
-        cast = 2.0,
+        cast = 0.0,
+        channeled = true,
         cooldown = 45.0,
         gcd = "global",
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': DAMAGE_FROM_MAX_HEALTH_PCT, 'subtype': NONE, 'points': 8.0, 'radius': 100.0, 'target': TARGET_UNIT_CASTER_AND_SUMMONS, }
     },
 
