--- "D:\\Development\\WoW Python Tools\\Output\\11.0.2.55763\\DeathKnightBlood.lua"	2024-07-21 09:25:51.296850600 -1000
+++ "D:\\Development\\WoW Python Tools\\Output\\11.0.5.56865\\DeathKnightBlood.lua"	2024-10-21 13:51:29.311509400 -1000
@@ -1,12 +1,12 @@
 -- DeathKnightBlood.lua
--- July 2024
+-- October 2024
 
 local addon, ns = ...
 local Hekili = _G[ addon ]
 local class, state = Hekili.Class, Hekili.State
 
 local spec = Hekili:NewSpecialization( 250 )
 
 -- Resources
 spec:RegisterResource( Enum.PowerType.RunicPower )
 spec:RegisterResource( Enum.PowerType.Runes )
@@ -45,111 +45,110 @@
     rune_mastery                   = { 76079, 374574, 2 }, -- Consuming a Rune has a chance to increase your Strength by $s1% for $374585d.
     runic_attenuation              = { 76045, 207104, 1 }, -- Auto attacks have a chance to generate ${$221322s1/10} Runic Power.
     runic_protection               = { 76055, 454788, 1 }, -- Your chance to be critically struck is reduced by $s2% and your Armor is increased by $s1%.
     sacrificial_pact               = { 76060, 327574, 1 }, -- Sacrifice your ghoul to deal $327611s1 Shadow damage to all nearby enemies and heal for $s1% of your maximum health. Deals reduced damage beyond $327611s2 targets.
     soul_reaper                    = { 76063, 343294, 1 }, -- Strike an enemy for $s1 Shadowfrost damage and afflict the enemy with Soul Reaper. ; After $d, if the target is below $s3% health this effect will explode dealing an additional $343295s1 Shadowfrost damage to the target. If the enemy that yields experience or honor dies while afflicted by Soul Reaper, gain Runic Corruption.
     subduing_grasp                 = { 76080, 454822, 1 }, -- When you pull an enemy, the damage they deal to you is reduced by $454824s1% for $454824d.
     suppression                    = { 76087, 374049, 1 }, -- Damage taken from area of effect attacks reduced by $s1%. When suffering a loss of control effect, this bonus is increased by an additional $454886s1% for $454886d.
     unholy_bond                    = { 76076, 374261, 1 }, -- Increases the effectiveness of your Runeforge effects by $s1%.
     unholy_endurance               = { 76058, 389682, 1 }, -- Increases Lichborne duration by ${$s1/1000} sec and while active damage taken is reduced by $49039s8%.
     unholy_ground                  = { 76069, 374265, 1 }, -- Gain $374271s1% Haste while you remain within your Death and Decay.
-    unyielding_will                = { 76050, 457574, 1 }, -- Anti-Magic Shell's cooldown is increased by ${$s2/1000} sec and it now also removes all harmful magic effects when activated.
+    unyielding_will                = { 76050, 457574, 1 }, -- Anti-Magic shell now removes all harmful magical effects when activated, but it's cooldown is increased by ${$s2/1000}  sec.
     vestigial_shell                = { 76053, 454851, 1 }, -- Casting Anti-Magic Shell grants $454863i nearby allies a Lesser Anti-Magic Shell that Absorbs up to $454863s1 magic damage and reduces the duration of harmful Magic effects against them by $454863s2%. 
     veteran_of_the_third_war       = { 76068, 48263 , 1 }, -- Stamina increased by $s1%.; $?s316714[Damage taken reduced by $s3%.][]
     will_of_the_necropolis         = { 76054, 206967, 2 }, -- Damage taken below $s3% Health is reduced by $s2%.
     wraith_walk                    = { 76077, 212552, 1 }, -- Embrace the power of the Shadowlands, removing all root effects and increasing your movement speed by $s1% for $d. Taking any action cancels the effect.; While active, your movement speed cannot be reduced below $m2%.
 
     -- Blood Talents
-    bind_in_darkness               = { 95043, 440031, 1 }, -- Shadowfrost damage applies 2 stacks to Reaper's Mark and 4 stacks when it is a critical strike.; Additionally, $?a137008[Blood Boil][Rime empowered Howling Blast] deals Shadowfrost damage.
+    bind_in_darkness               = { 95043, 440031, 1 }, -- $?a137008[Blood Boil deals $s3% increased damage][Rime empowered Howling Blast deals $s4% increased damage to its main target], and is now Shadowfrost.; Shadowfrost damage applies 2 stacks to Reaper's Mark and 4 stacks when it is a critical strike.
     blood_boil                     = { 76170, 50842 , 1 }, -- [55078] A shadowy disease that drains $o1 health from the target over $d.  
     blood_feast                    = { 102243, 391386, 1 }, -- Anti-Magic Shell heals you for $s1% of the damage it absorbs.
-    blood_fever                    = { 95058, 440002, 1 }, -- Your $?a137008[Blood Plague][Frost Fever] has a chance to deal $s1% increased damage as Shadowfrost.
     blood_tap                      = { 76039, 221699, 1 }, -- Consume the essence around you to generate $s1 Rune.; Recharge time reduced by $s2 sec whenever a Bone Shield charge is consumed.
     blooddrinker                   = { 102244, 206931, 1 }, -- Drains $o1 health from the target over $d. The damage they deal to you is reduced by $s2% for the duration and $458687d after channeling it fully.; You can move, parry, dodge, and use defensive abilities while channeling this ability.; Generates ${$s3*4/10} additional Runic Power over the duration.
     bloodied_blade                 = { 102242, 458753, 1 }, -- Parrying an attack grants you a charge of Bloodied Blade, increasing your Strength by ${$460499s1/10}.1%, up to ${$460499u*$460499s1/10}.1% for $460499d.; At $460499U stacks, your next parry consumes all charges to unleash a Heart Strike at $s1% effectiveness, and increases your Strength by $460500s1% for $460500d.
     bloodshot                      = { 76125, 391398, 1 }, -- While Blood Shield is active, you deal $s1% increased Physical damage.
     bloodsoaked_ground             = { 95048, 434033, 1 }, -- While you are within your Death and Decay, your physical damage taken is reduced by ${$434034s1*-1}% and your chance to gain Vampiric Strike is increased by $s2%.
     bloodworms                     = { 76174, 195679, 1 }, -- Your auto attacks have a chance to summon a Bloodworm.; Bloodworms deal minor damage to your target for $198494d and then burst, healing you for $s3% of your missing health.; If you drop below $s2% health, your Bloodworms will immediately burst and heal you.
     bloody_fortitude               = { 95056, 434136, 1 }, -- Icebound Fortitude reduces all damage you take by up to an additional $s1% based on your missing health.; Killing an enemy that yields experience or honor reduces the cooldown of Icebound Fortitude by $s2 sec.
     bone_collector                 = { 76171, 458572, 1 }, -- [195181] Surrounds you with a barrier of whirling bones, increasing Armor by ${$s1*$STR/100}$?s316746[, and your Haste by $s4%][]. Each melee attack against you consumes a charge. Lasts $d or until all charges are consumed.
-    bonestorm                      = { 76127, 194844, 1 }, -- Consume your Bone Shield charges to create a whirl of bone and gore that batters all nearby enemies, dealing $196528s1 Shadow damage every $t3 sec, and healing you for $196545s1% of your maximum health every time it deals damage (up to ${$s1*$s4}%). Deals reduced damage beyond $196528s2 targets.; Lasts $d per Bone Shield charge spent and rapidly regenerates a Bone Shield every $t3 sec.
+    bonestorm                      = { 76127, 194844, 1 }, -- Consume up to $s4 Bone Shield charges to create a whirl of bone and gore that batters all nearby enemies, dealing $196528s1 Shadow damage every $t3 sec, and healing you for $196545s1% of your maximum health every time it deals damage (up to ${$s1*$s4}%). Deals reduced damage beyond $196528s2 targets.; Lasts $d per Bone Shield charge spent and rapidly regenerates a Bone Shield every $t3 sec.
     carnage                        = { 102245, 458752, 1 }, -- Blooddrinker and Consumption now contribute to your Mastery: Blood Shield.; Each time an enemy strikes your Blood Shield, the cooldowns of Blooddrinker and Consumption have a chance to be reset.
     coagulopathy                   = { 76038, 391477, 1 }, -- Enemies affected by Blood Plague take $s1% increased damage from you and Death Strike increases the damage of your Blood Plague by $391481s1% for $391481d, stacking up to $391481u times.
     consumption                    = { 102244, 274156, 1 }, -- Strikes all enemies in front of you with a hungering attack that deals $sw1 Physical damage and heals you for ${$e1*100}% of that damage. Deals reduced damage beyond $s3 targets.; Causes your Blood Plague damage to occur $s5% more quickly for $d. ; Generates $s4 Runes.
     dancing_rune_weapon            = { 76138, 49028 , 1 }, -- Summons a rune weapon for $81256d that mirrors your melee attacks and bolsters your defenses.; While active, you gain $81256s1% parry chance.
-    dark_talons                    = { 95057, 436687, 1 }, -- $?a137008[Marrowrend and Heart Strike have][Consuming Killing Machine or Rime has] a $s1% chance to increase the maximum stacks of an active $@spellname194878 by $s2, up to $s3 times.; While Icy Talons is active, your Runic Power spending abilities also count as Shadowfrost damage.
+    dark_talons                    = { 95057, 436687, 1 }, -- $?a137008[Marrowrend and Heart Strike have][Consuming Killing Machine or Rime has] a $s1% chance to grant $s2 stacks of Icy Talons and increase its maximum stacks by the same amount for 6 sec.; Runic Power spending abilities count as Shadowfrost while Icy Talons is active.
     deaths_messenger               = { 95049, 437122, 1 }, -- Reduces the cooldowns of Lichborne and Raise Dead by ${$s1/-1000} sec.
     everlasting_bond               = { 76130, 377668, 1 }, -- Summons $s1 additional copy of Dancing Rune Weapon and increases its duration by ${$s2/1000} sec.
     expelling_shield               = { 95049, 439948, 1 }, -- When an enemy deals direct damage to your Anti-Magic Shell, their cast speed is reduced by $440739s1% for $440739d.
-    exterminate                    = { 95068, 441378, 1 }, -- After Reaper's Mark explodes, your next $?a443564[$s3 ][]$?a137008[Marrowrend][Obliterate]$?a443564[s][] costs $?a443564[$s4][no] Rune and summon$?a443564[][s] $s1 scythes to strike your enemies.; The first scythe strikes your target for $?a137008[$441424s1][$441424s2] Shadowfrost damage and has a $?a443564[${$s2*1.5}][$s2]% chance to apply Reaper's Mark, the second scythe strikes all enemies around your target for $?a137008[$441426s1][$441426s2] Shadowfrost damage$?(a441894&a137008)[ and applies Blood Plague]?(a441894&$a137006)[ and applies Frost Fever][]. Deals reduced damage beyond $s5 targets.
+    exterminate                    = { 95068, 441378, 1 }, -- After Reaper's Mark explodes, your next $s3 $?a137008[$LMarrowrend:Marrowrends;][$LObliterate:Obliterates;] cost $s4 $LRune:Runes; and summon $s1 $Lscythe:scythes; to strike your enemies.; The first scythe strikes your target for $?a137008[$441424s1][$441424s2] Shadowfrost damage and has a $s2% chance to apply Reaper's Mark, the second scythe strikes all enemies around your target for $?a137008[$441426s1][$441426s2] Shadowfrost damage$?(a441894&a137008)[ and applies Blood Plague]?(a441894&$a137006)[ and applies Frost Fever][]. Deals reduced damage beyond $s5 targets.
     foul_bulwark                   = { 76167, 206974, 1 }, -- Each charge of Bone Shield increases your maximum health by $<increase>%.
     frenzied_bloodthirst           = { 95065, 434075, 1 }, -- Essence of the Blood Queen stacks $s1 additional times and increases the damage of your Death Coil and Death Strike by $s2% per stack.
-    gift_of_the_sanlayn            = { 95053, 434152, 1 }, -- While Vampiric Blood or Dark Transformation is active you gain Gift of the San'layn.; Gift of the San'layn increases the effectiveness of your Essence of the Blood Queen by $?a137007[$434153s1][$434153s4]%, and Vampiric Strike replaces your $?a137008[Heart Strike]?s207311[Clawing Shadows][Scourge Strike] for the duration.; 
+    gift_of_the_sanlayn            = { 95053, 434152, 1 }, -- While $?a137008[Dancing Rune Weapon][Dark Transformation] is active you gain Gift of the San'layn.; Gift of the San'layn increases the effectiveness of your Essence of the Blood Queen by $?a137007[$434153s1][$434153s4]%, and Vampiric Strike replaces your $?a137008[Heart Strike]?s207311[Clawing Shadows][Scourge Strike] for the duration.; 
     gorefiends_grasp               = { 76042, 108199, 1 }, -- Shadowy tendrils coil around all enemies within $A2 yards of a hostile or friendly target, pulling them to the target's location.
-    grim_reaper                    = { 95034, 434905, 1 }, -- Reaper's Mark explosion deals up to $s1% increased damage based on your target's missing health, and applies Soul Reaper to targets below $s2% health.
+    grim_reaper                    = { 95034, 434905, 1 }, -- Reaper's Mark initial strike grants $?a137008][$s3 charges of Bone Shield][Killing Machine].; Reaper's Mark explosion deals up to $s1% increased damage based on your target's missing health.
     heart_strike                   = { 76169, 206930, 1 }, -- Instantly strike the target and 1 other nearby enemy, causing $s2 Physical damage, and reducing enemies' movement speed by $s5% for $d$?s316575[; Generates $s3 bonus Runic Power][]$?s221536[, plus ${$210738s1/10} Runic Power per additional enemy struck][].
     heartbreaker                   = { 76135, 221536, 1 }, -- Heart Strike generates ${$210738s1/10} additional Runic Power per target hit.
     heartrend                      = { 76131, 377655, 1 }, -- Heart Strike has a chance to increase the damage of your next Death Strike by $s1%.
     hemostasis                     = { 76137, 273946, 1 }, -- Each enemy hit by Blood Boil increases the damage and healing done by your next Death Strike by $273947s1%, stacking up to $273947u times.
     improved_bone_shield           = { 76142, 374715, 1 }, -- Bone Shield increases your Haste by $s1%.
     improved_heart_strike          = { 76126, 374717, 2 }, -- Heart Strike damage increased by $s1%.
     improved_vampiric_blood        = { 76140, 317133, 2 }, -- Vampiric Blood's healing and absorb amount is increased by $s1% and duration by ${$s3/1000} sec.
     incite_terror                  = { 95040, 434151, 1 }, -- Vampiric Strike and $?a137008[Heart Strike]?s207311[Clawing Shadows][Scourge Strike] cause your targets to take $458478s1% increased Shadow damage, up to ${$458478s1*$458478U}% for $458478d.; Vampiric Strike benefits from Incite Terror at $s2% effectiveness.
     infliction_of_sorrow           = { 95033, 434143, 1 }, -- When Vampiric Strike damages an enemy affected by your $?a137008[Blood Plague][Virulent Plague], it extends the duration of the disease by $s3 sec, and deals $s2% of the remaining damage to the enemy. ; After Gift of the San'layn ends, your next $?a137008[Heart Strike]?s207311[Clawing Shadows][Scourge Strike] consumes the disease to deal $s1% of their remaining damage to the target.
     insatiable_blade               = { 76129, 377637, 1 }, -- Dancing Rune Weapon generates $s2 Bone Shield charges. When a charge of Bone Shield is consumed, the cooldown of Dancing Rune Weapon is reduced by ${$s1/-1000} sec.
     iron_heart                     = { 76172, 391395, 1 }, -- Blood Shield's duration is increased by ${$s1/1000} sec and it absorbs $s2% more damage.
     leeching_strike                = { 76145, 377629, 1 }, -- Heart Strike heals you for ${$s1/100}.2% health for each enemy hit while affected by Blood Plague.
     mark_of_blood                  = { 76139, 206940, 1 }, -- Places a Mark of Blood on an enemy for $d. The enemy's damaging auto attacks will also heal their victim for $206940s1% of the victim's maximum health.
     marrowrend                     = { 76168, 195182, 1 }, -- [195181] Surrounds you with a barrier of whirling bones, increasing Armor by ${$s1*$STR/100}$?s316746[, and your Haste by $s4%][]. Each melee attack against you consumes a charge. Lasts $d or until all charges are consumed.
     newly_turned                   = { 95064, 433934, 1 }, -- Raise Ally revives players at full health and grants you and your ally an absorb shield equal to $s2% of your maximum health.
     ossified_vitriol               = { 76146, 458744, 1 }, -- When you lose a Bone Shield charge the damage of your next Marrowrend is increased by 15%, stacking up to 75%.
     ossuary                        = { 76144, 219786, 1 }, -- While you have at least $s1 Bone Shield charges, the cost of Death Strike is reduced by ${$219788m1/-10} Runic Power.; Additionally, your maximum Runic Power is increased by ${$m2/10}.
     pact_of_the_deathbringer       = { 95035, 440476, 1 }, -- When you suffer a damaging effect equal to $s1% of your maximum health, you instantly cast Death Pact at $s3% effectiveness. May only occur every $s2 min.; When a Reaper's Mark explodes, the cooldowns of this effect and Death Pact are reduced by $s4 sec.
     pact_of_the_sanlayn            = { 95055, 434261, 1 }, -- You store $s1% of all Shadow damage dealt into your Blood Beast to explode for additional damage when it expires.
-    painful_death                  = { 95032, 443564, 1 }, -- Reaper's Mark deals $s1% increased damage and Exterminate empowers an additional $?a137008[Marrowrend][Obliterate], but now reduces its cost by 1 Rune.; Additionally, Exterminate now has a $s2% chance to apply Reaper's Mark.
     perseverance_of_the_ebon_blade = { 76124, 374747, 1 }, -- When Crimson Scourge is consumed, you gain $374748s1% Versatility for $374748d.
     purgatory                      = { 76133, 114556, 1 }, -- An unholy pact that prevents fatal damage, instead absorbing incoming healing equal to the damage prevented, lasting $116888d.; If any healing absorption remains when this effect expires, you will die. This effect may only occur every $123981d.
     rapid_decomposition            = { 76141, 194662, 1 }, -- Your Blood Plague and Death and Decay deal damage $s2% more often.; Additionally, your Blood Plague leeches $s3% more Health.
+    reaper_of_souls                = { 95034, 440002, 1 }, -- When you apply Reaper's Mark, the cooldown of Soul Reaper is reset, your next Soul Reaper costs no runes, and it explodes on the target regardless of their health.; Soul Reaper damage is increased by $s1%.
     reapers_mark                   = { 95062, 439843, 1 }, -- Viciously slice into the soul of your enemy, dealing $?a137008[$s1][$s4] Shadowfrost damage and applying Reaper's Mark.; Each time you deal Shadow or Frost damage, add a stack of Reaper's Mark. After $434765d or reaching $434765u stacks, the mark explodes, dealing $?a137008[$436304s1][$436304s2] damage per stack.; Reaper's Mark travels to an unmarked enemy nearby if the target dies, or explodes below 35% health when there are no enemies to travel to. This explosion cannot occur again on a target for $443761d.
+    reapers_onslaught              = { 95057, 469870, 1 }, -- Reduces the cooldown of Reaper's Mark by ${$s1/-1000} sec, but the amount of $?a137008[Marrowrends][Obliterates] empowered by Exterminate is reduced by $s2.
     red_thirst                     = { 76132, 205723, 1 }, -- Reduces the cooldown on Vampiric Blood by ${$s1/1000}.1 sec per $s2 Runic Power spent.
     reinforced_bones               = { 76143, 374737, 1 }, -- Increases Armor gained from Bone Shield by $s1% and it can stack $s2 additional times.
     relish_in_blood                = { 76147, 317610, 1 }, -- While Crimson Scourge is active, your next Death and Decay heals you for $317614s1 health per Bone Shield charge and you immediately gain ${$317614s2/10} Runic Power.
-    rune_carved_plates             = { 95035, 440282, 1 }, -- Each Rune spent reduces the magic damage you take by $440290s1% and each Rune generated reduces the physical damage you take by $440289s1% for $440289d, up to $440290u times.
+    rune_carved_plates             = { 95035, 440282, 1 }, -- Each Rune spent reduces the magic damage you take by ${$abs($440290s1/10)}.1% and each Rune generated reduces the physical damage you take by ${$abs($440289s1/10)}.1% for $440289d, up to $440290u times.
     rune_tap                       = { 76166, 194679, 1 }, -- Reduces all damage taken by $s1% for $d.
     sanguine_ground                = { 76041, 391458, 1 }, -- You deal $391459s1% more damage and receive $391459s2% more healing while standing in your Death and Decay.
     sanguine_scent                 = { 95055, 434263, 1 }, -- Your Death Coil$?a137007[, Epidemic][] and Death Strike have a $s2% increased chance to trigger Vampiric Strike when damaging enemies below $s1% health.
     shattering_bone                = { 76128, 377640, 1 }, -- When Bone Shield is consumed it shatters dealing $s3 Shadow damage to nearby enemies. This damage is tripled while you are within your Death and Decay.
-    soul_rupture                   = { 95061, 437161, 1 }, -- When Reaper's Mark explodes, it deals $?a137008[$s1][$s2]% of the damage dealt damage to nearby enemies.; Enemies hit by this effect deal $s3% reduced physical damage to you for 10 sec.
-    swift_end                      = { 95032, 443560, 1 }, -- Reaper's Mark's cost is reduced by $s2 Rune and its cooldown is reduced by ${$s1/-1000} sec.
-    the_blood_is_life              = { 95046, 434260, 1 }, -- Vampiric Strike has a chance to summon a Blood Beast to attack your enemy for $434237d.; Each time the Blood Beast attacks, it stores a portion of the damage dealt. When the Blood Beast dies, it explodes, dealing $?a137007[$s2][$s1]% of the damage accumulated to nearby enemies and healing the Death Knight for the same amount.
+    soul_rupture                   = { 95061, 437161, 1 }, -- When Reaper's Mark explodes, it deals $?a137008[$s1][$s2]% of the damage dealt to nearby enemies and causes them to deal $s3% reduced Physical damage to you for 10 sec.
+    swift_and_painful              = { 95032, 443560, 1 }, -- If no enemies are struck by Soul Rupture, you gain $469169s1% Strength for $469169d.; Wave of Souls is $s2% more effective on the main target of your Reaper's Mark.
+    the_blood_is_life              = { 95046, 434260, 1 }, -- Vampiric Strike has a chance to summon a Blood Beast to attack your enemy for $434237d.; Each time the Blood Beast attacks, it stores a portion of the damage dealt. When the Blood Beast dies, it explodes, dealing $?a137007[$s2][$s1]% of the damage accumulated to nearby enemies and healing the Death Knight for the same amount. Deals reduced damage beyond $s3 targets.
     tightening_grasp               = { 76165, 206970, 1 }, -- Gorefiend's Grasp cooldown is reduced by ${$m1/-1000} sec and it now also Silences enemies for $374776d.
     tombstone                      = { 76139, 219809, 1 }, -- Consume up to $s5 Bone Shield charges. For each charge consumed, you gain $s3 Runic Power and absorb damage equal to $s4% of your maximum health for $d.
     umbilicus_eternus              = { 76040, 391517, 1 }, -- After Vampiric Blood expires, you absorb damage equal to $s1 times the damage your Blood Plague dealt during Vampiric Blood.
     vampiric_aura                  = { 95056, 434100, 1 }, -- Your Leech is increased by $s1%.; While Lichborne is active, the Leech bonus of this effect is increased by $434105s1%, and it affects $s2 allies within 12 yds. 
     vampiric_blood                 = { 76173, 55233 , 1 }, -- Embrace your undeath, increasing your maximum health by $s4% and increasing all healing and absorbs received by $s1% for $d.
     vampiric_speed                 = { 95064, 434028, 1 }, -- Death's Advance and Wraith Walk movement speed bonuses are increased by $s1%.; Activating Death's Advance or Wraith Walk increases $434029s2 nearby allies movement speed by $434029s1% for $434029d.
     vampiric_strike                = { 95051, 433901, 1 }, -- Your Death Coil$?a137007[, Epidemic][] and Death Strike have a $s1% chance to make your next $?a137008[Heart Strike]?s207311[Clawing Shadows][Scourge Strike] become Vampiric Strike.; Vampiric Strike heals you for $?a137008[$434422s2][$434422s3]% of your maximum health and grants you Essence of the Blood Queen, increasing your Haste by ${$433925s1/10}.1%, up to ${$433925s1*$433925u/10}.1% for $433925d.; 
     visceral_strength              = { 95045, 434157, 1 }, -- When $?a137008[Crimson Scourge][Sudden Doom] is consumed, you gain $?a137008[$461130s1][$434159s1]% Strength for $?a137008[$461130d][$434159d].
     voracious                      = { 76043, 273953, 1 }, -- Death Strike's healing is increased by $s2% and grants you $274009s1% Leech for $274009d.
     wave_of_souls                  = { 95036, 439851, 1 }, -- Reaper's Mark sends forth bursts of Shadowfrost energy and back, dealing $?a137008[$435802s1][$435802s2] Shadowfrost damage both ways to all enemies caught in its path.; Wave of Souls critical strikes cause enemies to take $443404s1% increased Shadowfrost damage for $443404d, stacking up to 2 times, and it is always a critical strike on its way back.
-    wither_away                    = { 95057, 441894, 1 }, -- $?a137008[Blood Plague][Frost Fever] deals its damage in half the duration and the second scythe of Exterminate applies $?a137008[Blood Plague][Frost Fever].
+    wither_away                    = { 95058, 441894, 1 }, -- $?a137008[Blood Plague][Frost Fever] deals its damage $?a137008[$s5][$s6]% faster, and the second scythe of Exterminate applies $?a137008[Blood Plague][Frost Fever].
 } )
 
 -- PvP Talents
 spec:RegisterPvpTalents( {
     bloodforged_armor = 5587, -- (410301) Death Strike reduces all Physical damage taken by $410305s1% for $410305d.
     dark_simulacrum   = 3511, -- (77606 ) Places a dark ward on an enemy player that persists for $d, triggering when the enemy next spends mana on a spell, and allowing the Death Knight to unleash an exact duplicate of that spell.
     death_chain       = 609 , -- (203173) Chains $x1 enemies together, dealing $m2 Shadow damage and causing $m1% of all damage taken to also be received by the others in the chain. Lasts for $d.
     decomposing_aura  = 3441, -- (199720) All enemies within $A2 yards slowly decay, losing up to $199721m1% of their max health every $m3 sec. Max $m2 stacks. Lasts 6 sec.
     last_dance        = 608 , -- (233412) Reduces the cooldown of Dancing Rune Weapon by $m1% and its duration by $m2%.
     murderous_intent  = 841 , -- (207018) [206891] You focus the assault on this target, increasing their damage taken by $s1% for $d. Each unique player that attacks the target increases the damage taken by an additional $s1%, stacking up to $u times.; Your melee attacks refresh the duration of Focused Assault.
-    necrotic_aura     = 5513, -- (199642) All enemies within $a2 yards take $214968m1% increased magical damage.
     rot_and_wither    = 204 , -- (202727) Your $?s315442[Death's Due][Death and Decay] rots enemies each time it deals damage, absorbing healing equal to $s1% of damage dealt.
     spellwarden       = 5592, -- (410320) Anti-Magic Shell is now usable on allies and its cooldown is reduced by ${$s1/-1000} sec.
     strangulate       = 206 , -- (47476 ) Shadowy tendrils constrict an enemy's throat, silencing them for $d$?s58618[ (${$d+($58618m1/1000)} sec when used on a target who is casting a spell)][].
     walking_dead      = 205 , -- (202731) Your Death Grip causes the target to be unable to move faster than normal movement speed for $212610d.
 } )
 
 -- Auras
 spec:RegisterAuras( {
     -- Recently pulled  by Abomination Limb and can't be pulled again.
     abomination_limb = {
@@ -203,43 +202,42 @@
         duration = 120.0,
         max_stack = 1,
 
         -- Affected by:
         -- blood_draw[374598] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
     -- Draining $w1 health from the target every $t1 sec.
     blood_plague = {
         id = 55078,
         duration = 24.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
-        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blood_death_knight[137008] #14: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blood_plague[55078] #3: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- frost_fever[55095] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- coagulopathy[391477] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
         -- consumption[274156] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'amplitude': 1.5, 'points': -30.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- rapid_decomposition[194662] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -15.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- rapid_decomposition[194662] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 50.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_TAKEN_BY_PCT, }
         -- wither_away[441894] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
         -- wither_away[441894] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- coagulopathy[391481] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 25.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- brittle[374557] #0: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- sanguine_ground[391459] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- sanguine_ground[391459] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
     -- Dealing $s1% reduced damage to $@auracaster.
     blooddrinker = {
         id = 458687,
         duration = 5.0,
         max_stack = 1,
     },
     -- Physical damage taken reduced by $w1%.
     bloodforged_armor = {
         id = 410305,
@@ -266,21 +264,21 @@
 
         -- Affected by:
         -- foul_bulwark[206974] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 1.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- improved_bone_shield[374715] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
         -- reinforced_bones[374737] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
         -- reinforced_bones[374737] #1: { 'type': APPLY_AREA_AURA_PARTY, 'subtype': ADD_FLAT_MODIFIER, 'points': 2.0, 'value': 37, 'schools': ['physical', 'fire', 'shadow'], 'target': TARGET_UNIT_CASTER, }
     },
     -- Dealing $196528s1 Shadow damage to nearby enemies every $t3 sec, and healing for $196545s1% of maximum health for each target hit (up to ${$s1*$s4}%).
     bonestorm = {
         id = 194844,
-        duration = 1.0,
+        duration = 2.0,
         max_stack = 1,
 
         -- Affected by:
         -- veteran_of_the_third_war[48263] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': THREAT, }
     },
     -- Damage taken from $@auracaster increased by $s1%.
     brittle = {
         id = 374557,
         duration = 5.0,
         max_stack = 1,
@@ -401,37 +399,36 @@
     frost_breath = {
         id = 190780,
         duration = 10.0,
         max_stack = 1,
     },
     -- Suffering $w1 Frost damage every $t1 sec.
     frost_fever = {
         id = 55095,
         duration = 24.0,
         tick_time = 3.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
-        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blood_plague[55078] #3: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- frost_fever[55095] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
-        -- wither_away[441894] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
-        -- wither_away[441894] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
+        -- wither_away[441894] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
+        -- wither_away[441894] #3: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- brittle[374557] #0: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- sanguine_ground[391459] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- sanguine_ground[391459] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
     -- Movement speed reduced by $s5%.
     heart_strike = {
         id = 206930,
         duration = 8.0,
         max_stack = 1,
 
         -- Affected by:
         -- blood_plague[55078] #2: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
@@ -496,26 +493,20 @@
         id = 391547,
         duration = 3.0,
         max_stack = 1,
     },
     -- Auto attacks will heal the victim for $206940s1% of their maximum health.
     mark_of_blood = {
         id = 206940,
         duration = 15.0,
         max_stack = 1,
     },
-    -- Taking $w1% increased magical damage.
-    necrotic_aura = {
-        id = 214968,
-        duration = 3.0,
-        max_stack = 1,
-    },
     -- Death Strike cost reduced by ${$m1/-10} Runic Power.
     ossuary = {
         id = 219788,
         duration = 3600,
         max_stack = 1,
     },
     -- Grants the ability to walk across water.
     path_of_frost = {
         id = 3714,
         duration = 600.0,
@@ -524,21 +515,21 @@
 
         -- Affected by:
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
     },
     -- Versatility increased by $w1%
     perseverance_of_the_ebon_blade = {
         id = 374748,
         duration = 6.0,
         max_stack = 1,
     },
-    -- You are a prey for the Deathbringer... This effect will explode for $436304s1 Shadowfrost damage for each stack.
+    -- Marked as prey for the Deathbringer. This effect will explode for $436304s1 Shadowfrost damage for each stack.
     reapers_mark = {
         id = 434765,
         duration = 12.0,
         tick_time = 1.0,
         max_stack = 1,
     },
     -- Magical damage taken reduced by $w1%.
     rune_carved_plates = {
         id = 440290,
         duration = 5.0,
@@ -564,49 +555,51 @@
         id = 391459,
         duration = 3600,
         max_stack = 1,
     },
     -- Absorbing the next ${$w1-1} healing received.; If any amount of heal absorption remains when this effect expires, you die.
     shroud_of_purgatory = {
         id = 116888,
         duration = 3.0,
         max_stack = 1,
     },
-    -- Afflicted by Soul Reaper, if the target is below $s3% health this effect will explode dealing an additional $343295s1 Shadowfrost damage.
+    -- Afflicted by Soul Reaper, this effect will explode dealing an additional $343295s1 Shadowfrost damage.
     soul_reaper = {
-        id = 343294,
+        id = 469180,
         duration = 5.0,
         tick_time = 5.0,
         max_stack = 1,
-
-        -- Affected by:
-        -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
-        -- unholy_death_knight[137007] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
     -- Silenced.
     strangulate = {
         id = 47476,
-        duration = 5.0,
+        duration = 4.0,
         max_stack = 1,
     },
     -- Damage dealt to $@auracaster reduced by $w1%.
     subduing_grasp = {
         id = 454824,
         duration = 6.0,
         max_stack = 1,
     },
     -- Damage taken from area of effect attacks reduced by an additional $w1%.
     suppression = {
         id = 454886,
         duration = 6.0,
         max_stack = 1,
     },
+    -- Strength increased by $w1%.
+    swift_and_painful = {
+        id = 469169,
+        duration = 8.0,
+        max_stack = 1,
+    },
     -- Silenced.
     tightening_grasp = {
         id = 374776,
         duration = 3.0,
         max_stack = 1,
     },
     -- Absorbing $w1 damage.
     tombstone = {
         id = 219809,
         duration = 8.0,
@@ -630,21 +623,20 @@
     -- Vampiric Aura's Leech amount increased by $s1% and is affecting $s2 nearby allies.
     vampiric_aura = {
         id = 434105,
         duration = 3600,
         max_stack = 1,
     },
     -- Maximum health increased by $s4%. Healing and absorbs received increased by $s1%.
     vampiric_blood = {
         id = 55233,
         duration = 10.0,
-        pandemic = true,
         max_stack = 1,
 
         -- Affected by:
         -- improved_vampiric_blood[317133] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
         -- improved_vampiric_blood[317133] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- improved_vampiric_blood[317133] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 2000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
     -- Movement speed increased by $w1%.
     vampiric_speed = {
         id = 434029,
@@ -792,61 +784,61 @@
     blood_boil = {
         id = 50842,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         talent = "blood_boil",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Area Effects Use Target Radius'], 'ap_bonus': 0.56304, 'variance': 0.05, 'radius': 10.0, 'target': TARGET_DEST_CASTER, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Area Effects Use Target Radius'], 'ap_bonus': 0.56304001808, 'variance': 0.05000000075, 'radius': 10.0, 'target': TARGET_DEST_CASTER, 'target2': TARGET_UNIT_DEST_AREA_ENEMY, }
 
         -- Affected by:
-        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blood_plague[55078] #3: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- frost_fever[55095] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- bind_in_darkness[440031] #1: { 'type': APPLY_AURA, 'subtype': MOD_ABILITY_SCHOOL_MASK, 'points': 60.0, 'value': 48, 'schools': ['frost', 'shadow'], 'target': TARGET_UNIT_CASTER, }
+        -- bind_in_darkness[440031] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- brittle[374557] #0: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- sanguine_ground[391459] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- sanguine_ground[391459] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
 
     -- Drains $o1 health from the target over $d. The damage they deal to you is reduced by $s2% for the duration and $458687d after channeling it fully.; You can move, parry, dodge, and use defensive abilities while channeling this ability.; Generates ${$s3*4/10} additional Runic Power over the duration.
     blooddrinker = {
         id = 206931,
-        cast = 3.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 30.0,
         gcd = "global",
 
         spend = 1,
         spendType = 'runes',
 
         spend = -100,
         spendType = 'runic_power',
 
         talent = "blooddrinker",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_LEECH, 'amplitude': 1.25, 'tick_time': 1.0, 'ap_bonus': 0.639, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': APPLY_AURA, 'subtype': PERIODIC_LEECH, 'amplitude': 1.25, 'tick_time': 1.0, 'ap_bonus': 0.63899999857, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_IGNORE_TARGET_RESIST, 'amplitude': 1.25, 'points': -15.0, 'value': 127, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': APPLY_AURA, 'subtype': PERIODIC_ENERGIZE, 'amplitude': 1.25, 'tick_time': 1.0, 'points': 50.0, 'value': 6, 'schools': ['holy', 'fire'], 'target': TARGET_UNIT_CASTER, }
     },
 
-    -- Consume your Bone Shield charges to create a whirl of bone and gore that batters all nearby enemies, dealing $196528s1 Shadow damage every $t3 sec, and healing you for $196545s1% of your maximum health every time it deals damage (up to ${$s1*$s4}%). Deals reduced damage beyond $196528s2 targets.; Lasts $d per Bone Shield charge spent and rapidly regenerates a Bone Shield every $t3 sec.
+    -- Consume up to $s4 Bone Shield charges to create a whirl of bone and gore that batters all nearby enemies, dealing $196528s1 Shadow damage every $t3 sec, and healing you for $196545s1% of your maximum health every time it deals damage (up to ${$s1*$s4}%). Deals reduced damage beyond $196528s2 targets.; Lasts $d per Bone Shield charge spent and rapidly regenerates a Bone Shield every $t3 sec.
     bonestorm = {
         id = 194844,
         cast = 0.0,
         cooldown = 60.0,
         gcd = "global",
 
         talent = "bonestorm",
         startsCombat = false,
 
         -- Effects:
@@ -909,21 +901,21 @@
     consumption_274156 = {
         id = 274156,
         cast = 0.0,
         cooldown = 30.0,
         gcd = "global",
 
         talent = "consumption_274156",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': HEALTH_LEECH, 'subtype': NONE, 'amplitude': 1.5, 'ap_bonus': 0.78624, 'radius': 8.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
+        -- #0: { 'type': HEALTH_LEECH, 'subtype': NONE, 'amplitude': 1.5, 'ap_bonus': 0.78623998165, 'radius': 8.0, 'target': TARGET_UNIT_CONE_CASTER_TO_DEST_ENEMY, }
         -- #1: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 274893, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, }
         -- #3: { 'type': ENERGIZE, 'subtype': NONE, 'points': 2.0, 'target': TARGET_UNIT_CASTER, 'resource': runes, }
         -- #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'amplitude': 1.5, 'points': -30.0, 'target': TARGET_UNIT_CASTER, 'modifies': AURA_PERIOD, }
 
         -- Affected by:
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
         from = "spec_talent",
     },
 
@@ -963,26 +955,26 @@
     },
 
     -- Summons a rune weapon for $81256d that mirrors your melee attacks and bolsters your defenses.; While active, you gain $81256s1% parry chance.
     dancing_rune_weapon = {
         id = 49028,
         cast = 0.0,
         cooldown = 120.0,
         gcd = "global",
 
         talent = "dancing_rune_weapon",
-        startsCombat = true,
+        startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': SUMMON, 'subtype': NONE, 'amplitude': 1.0, 'points': 1.0, 'value': 27893, 'schools': ['physical', 'fire', 'frost', 'shadow', 'arcane'], 'value1': 3242, 'radius': 2.0, 'target': TARGET_UNK_151, }
         -- #1: { 'type': APPLY_AURA, 'subtype': DUMMY, 'trigger_spell': 1206, 'points': 50707.0, 'target': TARGET_UNIT_CASTER, }
-        -- #2: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 10.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #2: { 'type': APPLY_AURA, 'subtype': DUMMY, 'points': 10.0, 'target': TARGET_UNIT_TARGET_ANY, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
 
         -- Affected by:
         -- everlasting_bond[377668] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 6000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
         -- last_dance[233412] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
         -- last_dance[233412] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -2.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_4_VALUE, }
     },
 
     -- Command the target to attack you.
     dark_command = {
@@ -1043,21 +1035,21 @@
         id = 203173,
         color = 'pvp_talent',
         cast = 0.0,
         cooldown = 30.0,
         gcd = "global",
 
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': DUMMY, 'chain_targets': 3, 'points': 20.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 3, 'ap_bonus': 0.29483998, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 3, 'ap_bonus': 0.29483997822, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- Fires a blast of unholy energy at the target$?a377580[ and $377580s2 additional nearby target][], causing $47632s1 Shadow damage to an enemy or healing an Undead ally for $47633s1 health.$?s390268[; Increases the duration of Dark Transformation by $390268s1 sec.][]
     death_coil = {
         id = 47541,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 300,
@@ -1103,74 +1095,75 @@
         -- #0: { 'type': SCRIPT_EFFECT, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- deaths_echo[356367] #1: { 'type': APPLY_AURA, 'subtype': MOD_MAX_CHARGES, 'points': 1.0, 'target': TARGET_UNIT_CASTER, }
         -- deaths_reach[276079] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': RANGE, }
     },
 
     -- Create a death pact that heals you for $s1% of your maximum health, but absorbs incoming healing equal to $s3% of your max health for $d.
     death_pact = {
         id = 48743,
-        cast = 0.0,
+        cast = 15.0,
+        channeled = true,
         cooldown = 120.0,
         gcd = "none",
 
         talent = "death_pact",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': HEAL_PCT, 'subtype': NONE, 'points': 50.0, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': APPLY_AURA, 'subtype': SCHOOL_HEAL_ABSORB, 'value': 127, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Focuses dark power into a strike$?s137006[ with both weapons, that deals a total of ${$s1+$66188s1}][ that deals $s1] Physical damage and heals you for ${$s2}.2% of all damage taken in the last $s4 sec, minimum ${$s3}.1% of maximum health.
     death_strike = {
         id = 49998,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
-        spend = 500,
+        spend = 450,
         spendType = 'runic_power',
 
         talent = "death_strike",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.464256, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.4642559886, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
-        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 40.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- blood_death_knight[137008] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 62.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- blood_death_knight[137008] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 139.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- blood_plague[55078] #3: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- frost_fever[55095] #1: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
         -- improved_death_strike[374277] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.5, 'points': 60.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- improved_death_strike[374277] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.5, 'points': 60.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- improved_death_strike[374277] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- voracious[273953] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_2_VALUE, }
         -- voracious[273953] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- blood_draw[454871] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -100.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
-        -- hemostasis[273947] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- hemostasis[273947] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 8.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- brittle[374557] #0: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'points': 6.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
         -- ossuary[219788] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': -50.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
         -- sanguine_ground[391459] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- sanguine_ground[391459] #2: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 6.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #1: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': -5.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #5: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
+        -- unholy_death_knight[137007] #6: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'pvp_multiplier': 0.0, 'target': TARGET_UNIT_CASTER, 'modifies': PERIODIC_DAMAGE_HEALING, }
     },
 
     -- For $d, your movement speed is increased by $s1%, you cannot be slowed below $s2% of normal speed, and you are immune to forced movement effects and knockbacks.; Passive: You cannot be slowed below $124285s1% of normal speed.
     deaths_advance = {
         id = 48265,
         cast = 0.0,
         cooldown = 1.0,
         gcd = "none",
 
         startsCombat = false,
@@ -1195,21 +1188,21 @@
 
         spend = 1,
         spendType = 'runes',
 
         spend = -100,
         spendType = 'runic_power',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.19017, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.19017000496, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': TRIGGER_MISSILE, 'subtype': NONE, 'trigger_spell': 55078, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, }
     },
 
     -- Shadowy tendrils coil around all enemies within $A2 yards of a hostile or friendly target, pulling them to the target's location.
     gorefiends_grasp = {
         id = 108199,
         cast = 0.0,
         cooldown = 120.0,
         gcd = "global",
@@ -1239,21 +1232,21 @@
         spend = -150,
         spendType = 'runic_power',
 
 
 
         talent = "heart_strike",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'chain_targets': 2, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 2, 'ap_bonus': 0.475042, 'pvp_multiplier': 1.2, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 2, 'ap_bonus': 0.47504198551, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, }
         -- #3: { 'type': DUMMY, 'subtype': NONE, }
         -- #4: { 'type': APPLY_AURA, 'subtype': MOD_DECREASE_SPEED, 'chain_targets': 2, 'mechanic': snared, 'points': -20.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- blood_plague[55078] #2: { 'type': APPLY_AURA, 'subtype': MOD_SPELL_DAMAGE_FROM_CASTER, 'target': TARGET_UNIT_TARGET_ANY, }
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
         -- improved_heart_strike[374717] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
 
@@ -1306,39 +1299,38 @@
         spendType = 'runes',
 
         spend = -200,
         spendType = 'runic_power',
 
         talent = "marrowrend",
         startsCombat = true,
 
         -- Effects:
         -- #0: { 'type': DUMMY, 'subtype': NONE, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ENEMY, }
-        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.63954, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #1: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'chain_targets': 1, 'ap_bonus': 0.63954001665, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, 'chain_targets': 1, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Smash the target's mind with cold, interrupting spellcasting and preventing any spell in that school from being cast for $d.
     mind_freeze = {
         id = 47528,
         cast = 0.0,
         cooldown = 15.0,
         gcd = "none",
 
         spendType = 'runic_power',
 
         talent = "mind_freeze",
         startsCombat = true,
-        interrupt = true,
 
         -- Effects:
         -- #0: { 'type': INTERRUPT_CAST, 'subtype': NONE, 'mechanic': interrupted, 'target': TARGET_UNIT_TARGET_ENEMY, }
     },
 
     -- [206891] You focus the assault on this target, increasing their damage taken by $s1% for $d. Each unique player that attacks the target increases the damage taken by an additional $s1%, stacking up to $u times.; Your melee attacks refresh the duration of Focused Assault.
     murderous_intent = {
         id = 207018,
         color = 'pvp_talent',
         cast = 0.0,
@@ -1400,77 +1392,77 @@
         cooldown = 45.0,
         gcd = "global",
 
         spend = 2,
         spendType = 'runes',
 
         talent = "reapers_mark",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Chain from Initial Target', 'Enforce Line Of Sight To Chain Targets'], 'ap_bonus': 0.8, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Chain from Initial Target', 'Enforce Line Of Sight To Chain Targets'], 'ap_bonus': 1.5, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 434765, 'value': 10, 'schools': ['holy', 'nature'], 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': ENERGIZE, 'subtype': NONE, 'points': 200.0, 'target': TARGET_UNIT_CASTER, 'resource': runic_power, }
-        -- #3: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Chain from Initial Target', 'Enforce Line Of Sight To Chain Targets'], 'ap_bonus': 1.5, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #3: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'attributes': ['Chain from Initial Target', 'Enforce Line Of Sight To Chain Targets'], 'ap_bonus': 3.20000004768, 'pvp_multiplier': 0.60000002384, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
-        -- painful_death[443564] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER_BY_LABEL, 'points': 10.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
-        -- swift_end[443560] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
-        -- swift_end[443560] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -1.0, 'target': TARGET_UNIT_CASTER, 'modifies': POWER_COST, }
+        -- reapers_onslaught[469870] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER_BY_LABEL, 'points': -15000.0, 'target': TARGET_UNIT_CASTER, 'modifies': COOLDOWN, }
     },
 
     -- Strike the target for $s1 Physical damage. This attack cannot be dodged, blocked, or parried.
     rune_strike = {
         id = 316239,
         cast = 0.0,
         cooldown = 0.0,
         gcd = "global",
 
         spend = 1,
         spendType = 'runes',
 
         spend = -100,
         spendType = 'runic_power',
 
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.6, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.60000002384, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
 
         -- Affected by:
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Reduces all damage taken by $s1% for $d.
     rune_tap = {
         id = 194679,
-        cast = 0.0,
+        cast = 4.0,
+        channeled = true,
         cooldown = 1.5,
         gcd = "none",
 
         spend = 1,
         spendType = 'runes',
 
         spend = -100,
         spendType = 'runic_power',
 
         talent = "rune_tap",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': MOD_DAMAGE_PERCENT_TAKEN, 'points': -20.0, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_CASTER, }
     },
 
     -- Sacrifice your ghoul to deal $327611s1 Shadow damage to all nearby enemies and heal for $s1% of your maximum health. Deals reduced damage beyond $327611s2 targets.
     sacrificial_pact = {
         id = 327574,
         cast = 0.0,
+        channeled = true,
         cooldown = 120.0,
         gcd = "global",
 
         spend = 200,
         spendType = 'runic_power',
 
         talent = "sacrificial_pact",
         startsCombat = false,
 
         -- Effects:
@@ -1501,26 +1493,28 @@
         spend = 1,
         spendType = 'runes',
 
         spend = -100,
         spendType = 'runic_power',
 
         talent = "soul_reaper",
         startsCombat = true,
 
         -- Effects:
-        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.4488, 'variance': 0.05, 'target': TARGET_UNIT_TARGET_ENEMY, }
+        -- #0: { 'type': SCHOOL_DAMAGE, 'subtype': NONE, 'ap_bonus': 0.44879999757, 'variance': 0.05000000075, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #1: { 'type': APPLY_AURA, 'subtype': PERIODIC_DUMMY, 'tick_time': 5.0, 'target': TARGET_UNIT_TARGET_ENEMY, }
         -- #2: { 'type': DUMMY, 'subtype': NONE, }
+        -- #3: { 'type': TRIGGER_SPELL, 'subtype': NONE, 'trigger_spell': 469180, 'points': 35.0, }
 
         -- Affected by:
         -- death_knight[137005] #1: { 'type': APPLY_AURA, 'subtype': MOD_GLOBAL_COOLDOWN_BY_HASTE_REGEN, 'sp_bonus': 0.25, 'points': 100.0, 'value': 11, 'schools': ['physical', 'holy', 'nature'], 'target': TARGET_UNIT_CASTER, }
+        -- reaper_of_souls[440002] #0: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 20.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
         -- unholy_death_knight[137007] #4: { 'type': APPLY_AURA, 'subtype': ADD_PCT_MODIFIER, 'points': 15.0, 'target': TARGET_UNIT_CASTER, 'modifies': DAMAGE_HEALING, }
     },
 
     -- Shadowy tendrils constrict an enemy's throat, silencing them for $d$?s58618[ (${$d+($58618m1/1000)} sec when used on a target who is casting a spell)][].
     strangulate = {
         id = 47476,
         color = 'pvp_talent',
         cast = 0.0,
         cooldown = 45.0,
         gcd = "none",
@@ -1571,22 +1565,21 @@
 
         -- Affected by:
         -- improved_vampiric_blood[317133] #0: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_1_VALUE, }
         -- improved_vampiric_blood[317133] #1: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 5.0, 'target': TARGET_UNIT_CASTER, 'modifies': EFFECT_3_VALUE, }
         -- improved_vampiric_blood[317133] #2: { 'type': APPLY_AURA, 'subtype': ADD_FLAT_MODIFIER, 'points': 2000.0, 'target': TARGET_UNIT_CASTER, 'modifies': BUFF_DURATION, }
     },
 
     -- Embrace the power of the Shadowlands, removing all root effects and increasing your movement speed by $s1% for $d. Taking any action cancels the effect.; While active, your movement speed cannot be reduced below $m2%.
     wraith_walk = {
         id = 212552,
-        cast = 4.0,
-        channeled = true,
+        cast = 0.0,
         cooldown = 60.0,
         gcd = "global",
 
         talent = "wraith_walk",
         startsCombat = false,
 
         -- Effects:
         -- #0: { 'type': APPLY_AURA, 'subtype': MOD_INCREASE_SPEED, 'points': 70.0, 'target': TARGET_UNIT_CASTER, }
         -- #1: { 'type': APPLY_AURA, 'subtype': MOD_MINIMUM_SPEED, 'amplitude': 1.0, 'points': 170.0, 'target': TARGET_UNIT_CASTER, }
         -- #2: { 'type': APPLY_AURA, 'subtype': MOD_DAMAGE_PERCENT_TAKEN, 'schools': ['physical', 'holy', 'fire', 'nature', 'frost', 'shadow', 'arcane'], 'target': TARGET_UNIT_CASTER, }
